---
title: "analyse 2 Value branche ep = 0.006"
author: "Maxime"
date: "2025-04-27"
output: pdf_document
---



```{r}
library("reshape2")

library(ggplot2)
library(tidyr)
library(dplyr)
library(ggpubr)
library(reshape2)
```


```{r}
setwd("~/Documents/Darwin2/M2/S2/Stage/Stage gBC/tab_sous_ech_hydromyini.nosync")
```

longueur de branches inférieurs à 0.006




####Hydro à 15 sp
```{r}


##Sous écha = 15 sp

tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
#tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)
```

```{r}
hist(tab_ep_ponctual$post1,
     main = "Histogramme post 1",
     xlab = "post 1",
     ylab = "Fréquence",
     col = "lightblue",
     border = "white",
     breaks = 350)
```




```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_15sp/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)



# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)

#moy_NS_WS_obs = mean(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)
#moy_NS_WS_att <- as.data.frame(tab_random_NS_WS_postepNS[, 18:1017])
#mean(moy_NS_WS_att)


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_15sp/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_15sp/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_15sp/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_15sp/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_15sp/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_15sp/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_15sp/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual










nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_15 <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_15 <- melt(df_Z_scores_15, varnames = c("Condition", "Transition"))
df_points_15$Condition <- rep(rownames(df_Z_scores_15), times = ncol(df_Z_scores_15))

plot_15_sp = ggplot(df_points_15, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_15_sp


```








####Hydro à 25 sp

```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_25sp/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)



# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)

#moy_NS_WS_obs = mean(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)
#moy_NS_WS_att <- as.data.frame(tab_random_NS_WS_postepNS[, 18:1017])
#mean(moy_NS_WS_att)


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_25sp/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_25sp/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_25sp/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_25sp/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_25sp/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_25sp/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_25sp/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual










nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_25 <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_25 <- melt(df_Z_scores_25, varnames = c("Condition", "Transition"))
df_points_25$Condition <- rep(rownames(df_Z_scores_25), times = ncol(df_Z_scores_25))

plot_25_sp = ggplot(df_points_25, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_25_sp


```
















####Hydro à 37 sp


```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_37sp/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)



# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)

#moy_NS_WS_obs = mean(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)
#moy_NS_WS_att <- as.data.frame(tab_random_NS_WS_postepNS[, 18:1017])
#mean(moy_NS_WS_att)


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_37sp/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_37sp/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_37sp/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_37sp/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_37sp/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_37sp/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_37sp/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual










nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_37 <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_37 <- melt(df_Z_scores_37, varnames = c("Condition", "Transition"))
df_points_37$Condition <- rep(rownames(df_Z_scores_37), times = ncol(df_Z_scores_37))

plot_37_sp = ggplot(df_points_37, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_37_sp


```
















####Hydro à 48 sp


```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_48sp/tab_NS_WS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_48sp/tab_NS_SW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_48sp/tab_NS_SS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_48sp/tab_NS_WW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

	## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_48sp/tab_NS_WS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_48sp/tab_NS_SW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit


# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_48sp/tab_NS_SS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_48sp/tab_NS_WW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

	## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual












nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_48 <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_48 <- melt(df_Z_scores_48, varnames = c("Condition", "Transition"))
df_points_48$Condition <- rep(rownames(df_Z_scores_48), times = ncol(df_Z_scores_48))

plot_48_sp=ggplot(df_points_48, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_48_sp
```




####Rattini



```{r}
tab_tri_ep <- read.csv("Rattini/posterior_optimum_Rattini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("Rattini/paralog_exons_Rattini.txt", header=F)
exons_aln_error <- read.table("Rattini/List_exons_error_aln_Rattini.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

#tab_br_leng <- tab[, c("Exon_ep", "Br_lg", "Br_ep")]


#tab_ep_ponctual = 
# merge the table of punctual episodes with the table of informations on episodes
#tab_ep_ponctual <- merge(tab_br_leng, tab_ep_ponctual, by=c("Exon_ep","Br_ep"))

#tab_ep_ponctual = subset(tab_ep_ponctual, tab_ep_ponctual$Br_lg < 0.002)


	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

###récupère les tableaux des épisodes ponctuels, deux branches et tous les exons



## test for compensation ## FIGURE 5 ##
# NS after episodes with at least 1 NS-WS #

## WS


# load table
tab_random_NS_WS_postepNS <- read.csv("Rattini/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
#j'ai changé le nom des colonnes, j'ai enlevé le dn/ds
colnames(tab_random_NS_WS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

#colnames(tab_ep_ponctual)
#colnames(tab_random_NS_WS_postepNS)

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]
###j'ai mis un = pour dire qu'on garde quand on a un épisode 
#tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)
#tab_ep_ponctual = subset(tab_ep_ponctual, tab_ep_ponctual$Br_lg < 0.002)

#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("Rattini/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_SW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]
#tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)


# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit
	


#SS

# load table
tab_random_NS_SS_postepNS <- read.csv("Rattini/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

#colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br_ep', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'))
colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")


# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]
#tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 17:1016]))

colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")
	


#WW

# load table
tab_random_NS_WW_postepNS <- read.csv("Rattini/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_WW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")



# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]
#tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit




## total NS = WS+SW+SSWW

# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit


	## plots


# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual

# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS


# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("Rattini/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit



## SW




# load table
tab_random_NS_SW_postepnoNS <- read.csv("Rattini/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit



## SS



# load table
tab_random_NS_SS_postepnoNS <- read.csv("Rattini/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")



## WW




# load table
tab_random_NS_WW_postepnoNS <- read.csv("Rattini/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons


tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit


## total NS = WS+SW+SSWW


# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit



## plots


# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual







nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Rattini <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Rattini <- melt(df_Z_scores_Rattini, varnames = c("Condition", "Transition"))
df_points_Rattini$Condition <- rep(rownames(df_Z_scores_Rattini), times = ncol(df_Z_scores_Rattini))

plot_Rattini = ggplot(df_points_Rattini, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Rattini

```












####Cétacés 


```{r}
tab_tri_ep <- read.csv("Cetacea/posterior_optimum_Cetacea.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
#exons_paralog <- read.csv("paralog_exons.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
#tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

#tab_br_leng <- tab[, c("Exon_ep", "Br_lg", "Br_ep")]


#tab_ep_ponctual = 
# merge the table of punctual episodes with the table of informations on episodes
#tab_ep_ponctual <- merge(tab_br_leng, tab_ep_ponctual, by=c("Exon_ep","Br_ep"))

#tab_ep_ponctual = subset(tab_ep_ponctual, tab_ep_ponctual$Br_lg < 0.002)


	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

###récupère les tableaux des épisodes ponctuels, deux branches et tous les exons



## test for compensation ## FIGURE 5 ##
# NS after episodes with at least 1 NS-WS #

## WS


# load table
tab_random_NS_WS_postepNS <- read.csv("Cetacea/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
#j'ai changé le nom des colonnes, j'ai enlevé le dn/ds
colnames(tab_random_NS_WS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

#colnames(tab_ep_ponctual)
#colnames(tab_random_NS_WS_postepNS)

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]
###j'ai mis un = pour dire qu'on garde quand on a un épisode 
#tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)
#tab_ep_ponctual = subset(tab_ep_ponctual, tab_ep_ponctual$Br_lg < 0.002)

#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit



## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("Cetacea/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_SW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]
#tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)


# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit
	


#SS

# load table
tab_random_NS_SS_postepNS <- read.csv("Cetacea/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

#colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br_ep', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'))
colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")


# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]
#tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 17:1016]))

colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")
	


#WW

# load table
tab_random_NS_WW_postepNS <- read.csv("Cetacea/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_WW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")



# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]
#tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit




## total NS = WS+SW+SSWW

# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit


	## plots


# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual

# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS


# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("Cetacea/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit



## SW




# load table
tab_random_NS_SW_postepnoNS <- read.csv("Cetacea/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit



## SS



# load table
tab_random_NS_SS_postepnoNS <- read.csv("Cetacea/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")



## WW




# load table
tab_random_NS_WW_postepnoNS <- read.csv("Cetacea/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons


tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW


# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit



## plots


# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual









nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Cetacea <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Cetacea <- melt(df_Z_scores_Cetacea, varnames = c("Condition", "Transition"))
df_points_Cetacea$Condition <- rep(rownames(df_Z_scores_Cetacea), times = ncol(df_Z_scores_Cetacea))

plot_Cetacea = ggplot(df_points_Cetacea, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Cetacea










nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Cetacea <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Cetacea <- melt(df_Z_scores_Cetacea, varnames = c("Condition", "Transition"))
df_points_Cetacea$Condition <- rep(rownames(df_Z_scores_Cetacea), times = ncol(df_Z_scores_Cetacea))

plot_Cetacea = ggplot(df_points_Cetacea, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Cetacea

```














####Cétacés_25


```{r}
tab_tri_ep <- read.csv("Cetacea_25/posterior_optimum_Cetacea.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("Cetacea_25/paralog_exons.txt", header=F)
exons_aln_error <- read.table("Cetacea_25/List_exons_error_aln_old.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

#tab_br_leng <- tab[, c("Exon_ep", "Br_lg", "Br_ep")]

hist(tab_tri_ep$post1)
#tab_ep_ponctual = 
# merge the table of punctual episodes with the table of informations on episodes
#tab_ep_ponctual <- merge(tab_br_leng, tab_ep_ponctual, by=c("Exon_ep","Br_ep"))

#tab_ep_ponctual = subset(tab_ep_ponctual, tab_ep_ponctual$Br_lg < 0.002)


	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

###récupère les tableaux des épisodes ponctuels, deux branches et tous les exons



## test for compensation ## FIGURE 5 ##
# NS after episodes with at least 1 NS-WS #

## WS


# load table
tab_random_NS_WS_postepNS <- read.csv("Cetacea_25/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
#j'ai changé le nom des colonnes, j'ai enlevé le dn/ds
colnames(tab_random_NS_WS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

#colnames(tab_ep_ponctual)
#colnames(tab_random_NS_WS_postepNS)

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]
###j'ai mis un = pour dire qu'on garde quand on a un épisode 
#tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual

tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)
#tab_ep_ponctual = subset(tab_ep_ponctual, tab_ep_ponctual$Br_lg < 0.002)

#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit



## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("Cetacea_25/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_SW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]
#tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)


# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit
	


#SS

# load table
tab_random_NS_SS_postepNS <- read.csv("Cetacea_25/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

#colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br_ep', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'))
colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")


# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]
#tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 17:1016]))

colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")
	


#WW

# load table
tab_random_NS_WW_postepNS <- read.csv("Cetacea_25/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_WW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")



# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]
#tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit




## total NS = WS+SW+SSWW

# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit


	## plots


# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual

# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS


# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("Cetacea_25/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit



## SW




# load table
tab_random_NS_SW_postepnoNS <- read.csv("Cetacea_25/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit



## SS



# load table
tab_random_NS_SS_postepnoNS <- read.csv("Cetacea_25/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")



## WW




# load table
tab_random_NS_WW_postepnoNS <- read.csv("Cetacea_25/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons


tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg < 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW


# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit



## plots


# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual









nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Cetacea_25 <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Cetacea_25 <- melt(df_Z_scores_Cetacea_25, varnames = c("Condition", "Transition"))
df_points_Cetacea_25$Condition <- rep(rownames(df_Z_scores_Cetacea_25), times = ncol(df_Z_scores_Cetacea_25))

plot_Cetacea_25 = ggplot(df_points_Cetacea_25, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Cetacea_25

```










#MURINI
#####Récupérer valeurs lg branches etc Murini
```{r}
tab <- read.csv("tab_recap_all_genes_Murini.csv", header=FALSE, sep="\t")
colnames(tab) = c("Exon_ep","exon_pos", "Dn/Ds", "Lg_seq","GC12","GC3","Br_ep","Br_lg","Br_Asc","Br_Desc1","Br_Desc2","Episode","EpisodeSW","EpisodeWSSW","Episode_wtI","Nb_subst_S_WS","signif_nb_subst_S_WS","Nb_subst_S_SW","Nb_subst_S_WW","Nb_subst_S_SS","Nb_subst_NS_WS","Nb_subst_NS_SW","Nb_subst_NS_SS","Nb_subst_NS_WW")
###tableau avec des infos sur la branche concernée et tout 


tab <- tab[tab$Br_lg > 0, ]
###enlève les branches ayant une longueur nulle

# Extraire les exons uniques et les mettre en data frame
liste_exon <- unique(tab$Exon_ep)
# Compter le nombre d'exons uniques
nb_exons <- length(liste_exon)


# retrieve paralog exons and error align
exons_paralog <- read.csv("paralog_exons_murini.txt", header=F)
#exons_aln_error <- read.table("List_exons_error_aln_old.csv", header=T)

tab_sans_paralog <- tab[!tab$Exon %in% exons_paralog$V1, ]
#tab_sans_paralog <- tab[!tab$Exon %in% exons_aln_error$Exon, ]
###on vire les exons détectés comme paralogues 


# add GC-content
tab_sans_paralog$AT12 <- 100 - tab_sans_paralog$GC12
tab_sans_paralog$AT3 <- 100 - tab_sans_paralog$GC3
###on récupère le AT12 et le AT3


## number of episodes
tab_ep <- tab_sans_paralog[tab_sans_paralog$Episode == 'YES', ]
nb <- nrow(tab_ep)
nb
###récupère le nombre d'épisode 

# in internal branches (=episodes after which we can search for compensation)
tab_ep_sans_term <- tab_ep[!is.na(tab_ep$Br_Desc1), ]
tab_ep_br_int <- tab_ep_sans_term[!is.na(tab_ep_sans_term$Br_Asc), ]
nb <- nrow(tab_ep_br_int)
nb
###récupère les épisodes avec des branches internes 

# with at least 1 NS-WS substitution
	
tab_ep_NS_WS <- tab_ep[tab_ep$Nb_subst_NS_WS > 0.9, ]
nb <- nrow(tab_ep_NS_WS)
nb
###nombre d'épisodes avec au moins une subst NS WS dans la branche épisode 

tab_ep_NS_WS_br_int <- tab_ep_br_int[tab_ep_br_int$Nb_subst_NS_WS > 0.9, ]
nb <- nrow(tab_ep_NS_WS_br_int)
nb
#au moins une subst NS et c'est une branche interne 

## distributions of exon length, branch length, GC12 and GC3 ## FIGURE S1 ##

# in all exons
tab_exon_info <- tab_sans_paralog[, c("Exon_ep", "Lg_seq", "Br_lg", "GC3", "GC12")]
tab_exon_lg_seq <- distinct(tab_exon_info, Exon_ep, Lg_seq)
tab_exon_GC3 <- distinct(tab_exon_info, Exon_ep, GC3, GC12)

###fais juste des tabbleaux avec la colonne exon et ce qui nous intéresse 


####plot de la longueur moyenne des exons 
moy_lg_seq_all <- mean(tab_exon_lg_seq$Lg_seq)
plot_distrib_lg_seq_all <- ggplot(tab_exon_lg_seq, aes(Lg_seq)) +
  geom_histogram(binwidth=50, fill="azure3", color="black", size=0.3) +
  xlim(0,1000) +
  theme_linedraw() +
  xlab("") +
  ylab("counts") +
  geom_vline(xintercept = moy_lg_seq_all, size=0.8)
plot_distrib_lg_seq_all


####plot des longueurs de branches 
moy_br_lg_all <- mean(tab_exon_info$Br_lg)
moy_het <- 0.0026
plot_distrib_br_lg_all <- ggplot(tab_exon_info, aes(Br_lg)) +
  geom_histogram(fill="azure3", color="black", size=0.3, binwidth = 0.001) +
  theme_linedraw() +
  xlab("Longueur des branches") +
  ylab("") +
  geom_vline(xintercept = moy_br_lg_all, size=0.8)#+
  #geom_vline(xintercept = moy_het, size=0.8, linetype = "longdash")
plot_distrib_br_lg_all
moy_br_lg_all
#plot du GC3
moy_GC3_all <- mean(tab_exon_GC3$GC3)
plot_distrib_GC3_all <- ggplot(tab_exon_GC3, aes(GC3)) +
  geom_histogram(fill="azure3", color="black", size=0.3, binwidth=5) +
  theme_linedraw() +
  xlab("") +
  ylab("") +
  geom_vline(xintercept = moy_GC3_all, size=0.8) +
  xlim(0,100)
plot_distrib_GC3_all


#plot du GC12
moy_GC12_all <- mean(tab_exon_GC3$GC12)
plot_distrib_GC12_all <- ggplot(tab_exon_GC3, aes(GC12)) +
  geom_histogram(fill="azure3", color="black", size=0.3, binwidth=5) +
  theme_linedraw() +
  xlab("") +
  ylab("") +
  geom_vline(xintercept = moy_GC12_all, size=0.8) +
  xlim(0,100)
plot_distrib_GC12_all

plot_distrib_all <- ggarrange(plot_distrib_lg_seq_all, plot_distrib_br_lg_all, plot_distrib_GC3_all, plot_distrib_GC12_all, ncol = 4)
plot_distrib_all

# in exons with gBGC episodes

tab_exon_ep_info <- tab_ep[, c("Exon_ep", "Lg_seq", "Br_lg", "GC3", "GC12")]
tab_exon_ep_lg_seq <- distinct(tab_exon_ep_info, Exon_ep, Lg_seq)
tab_exon_ep_GC123 <- distinct(tab_exon_ep_info, Exon_ep, GC3, GC12)

moy_lg_seq_ep <- mean(tab_exon_ep_lg_seq$Lg_seq)
plot_distrib_lg_seq_ep <- ggplot(tab_exon_ep_lg_seq, aes(Lg_seq)) +
  geom_histogram(binwidth=50, fill="coral2", color="black", size=0.3) +
  xlim(0,1000) +
  theme_linedraw() +
  xlab("exon lengths") +
  ylab("counts") +
  geom_vline(xintercept = moy_lg_seq_ep, size=0.8)
plot_distrib_lg_seq_ep

moy_br_lg_ep <- mean(tab_exon_ep_info$Br_lg)
mean_het <- 0.0030
plot_distrib_br_lg_ep <- ggplot(tab_exon_ep_info, aes(Br_lg)) +
  geom_histogram(fill="coral2", color="black", size=0.3, binwidth=0.001) +
  theme_linedraw() +
  xlab("Longueur des branches") +
  ylab("") +
  geom_vline(xintercept = moy_br_lg_ep, size=0.8) #+
  #geom_vline(xintercept = mean_het, size=0.8, linetype = "longdash")
plot_distrib_br_lg_ep



moy_GC3_ep <- mean(tab_exon_ep_GC123$GC3)
plot_distrib_GC3_ep <- ggplot(tab_exon_ep_GC123, aes(GC3)) +
  geom_histogram(fill="coral2", color="black", size=0.3, binwidth=5) +
  theme_linedraw() +
  xlab("GC3") +
  ylab("") +
  geom_vline(xintercept = moy_GC3_ep, size=0.8) +
  xlim(0,100)
plot_distrib_GC3_ep

moy_GC12_ep <- mean(tab_exon_ep_GC123$GC12)
plot_distrib_GC12_ep <- ggplot(tab_exon_ep_GC123, aes(GC12)) +
  geom_histogram(fill="coral2", color="black", size=0.3, binwidth=5) +
  theme_linedraw() +
  xlab("GC12") +
  ylab("") +
  geom_vline(xintercept = moy_GC12_ep, size=0.8) +
  xlim(0,100)
plot_distrib_GC12_ep

plot_distrib_ep <- ggarrange(plot_distrib_lg_seq_ep, plot_distrib_br_lg_ep, plot_distrib_GC3_ep, plot_distrib_GC12_ep, ncol=4)
plot_distrib_ep

# final plot with two types of exons

plot_distrib_all_plus_ep <- ggarrange(plot_distrib_all, plot_distrib_ep, nrow=2, labels=c("A", "B"))
plot_distrib_all_plus_ep
```


#####Sélection compensatoire Murini
```{r}

tab_tri_ep <- read.csv("posterior_episodes_optimum_murini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_murini.txt", header=F)
#exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
#tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
#tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

hist(tab_ep_ponctual$post1,
     main = "Histogramme post 1",
     xlab = "post 1",
     ylab = "Fréquence",
     col = "lightblue",
     border = "white",
     breaks = 350)






## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("tab_all_genes_murini/tab_NS_WS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("tab_all_genes_murini/tab_NS_SW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_SW_postepNS = subset(tab_random_NS_SW_postepNS, tab_random_NS_SW_postepNS$Br_asc_lg < 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("tab_all_genes_murini/tab_NS_SS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_SS_postepNS = subset(tab_random_NS_SS_postepNS, tab_random_NS_SS_postepNS$Br_asc_lg > 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("tab_all_genes_murini/tab_NS_WW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_WW_postepNS = subset(tab_random_NS_WW_postepNS, tab_random_NS_WW_postepNS$Br_asc_lg > 0.006)



# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

	## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_WS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_WS_postepnoNS = subset(tab_random_NS_WS_postepnoNS, tab_random_NS_WS_postepnoNS$Br_asc_lg > 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_SW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_SW_postepnoNS = subset(tab_random_NS_SW_postepnoNS, tab_random_NS_SW_postepnoNS$Br_asc_lg > 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit


# load table
tab_random_NS_SS_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_SS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_SS_postepnoNS = subset(tab_random_NS_SS_postepnoNS, tab_random_NS_SS_postepnoNS$Br_asc_lg > 0.006)

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_WW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_WW_postepnoNS = subset(tab_random_NS_WW_postepnoNS, tab_random_NS_WW_postepnoNS$Br_asc_lg > 0.006)


# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

	## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual












nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Murini <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Murini <- melt(df_Z_scores_Murini, varnames = c("Condition", "Transition"))
df_points_Murini$Condition <- rep(rownames(df_Z_scores_Murini), times = ncol(df_Z_scores_Murini))

plot_Murini = ggplot(df_points_Murini, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Murini







```






###PLACER LES RATTINI + CETACÉS + MURINI SUR LE PLOT DES SUBSTITUTIONS 

```{r}


tab_recap_all_genes_Hydro = read.csv("tab_recap_all_genes.csv", header = F, sep ="\t")
colnames(tab_recap_all_genes_Hydro) <- c("Exon_ep","exon_pos", "Lg_seq","GC12","GC3","Br_ep","Br_lg","Br_Asc","Br_Desc1","Br_Desc2","Episode","EpisodeSW","EpisodeWSSW","Episode_wtI","Nb_subst_S_WS","signif_nb_subst_S_WS","Nb_subst_S_SW","Nb_subst_S_WW","Nb_subst_S_SS","Nb_subst_NS_WS","Nb_subst_NS_SW","Nb_subst_NS_SS","Nb_subst_NS_WW")

nb_subst_S_WS_all = sum(tab_recap_all_genes_Hydro$Nb_subst_S_WS)
nb_subst_S_SW_all = sum(tab_recap_all_genes_Hydro$Nb_subst_S_SW)
nb_subst_S_WW_all = sum(tab_recap_all_genes_Hydro$Nb_subst_S_WW)
nb_subst_S_SS_all = sum(tab_recap_all_genes_Hydro$Nb_subst_S_SS)

nb_subst_NS_WS_all = sum(tab_recap_all_genes_Hydro$Nb_subst_NS_WS)
nb_subst_NS_SW_all = sum(tab_recap_all_genes_Hydro$Nb_subst_NS_SW)
nb_subst_NS_WW_all = sum(tab_recap_all_genes_Hydro$Nb_subst_NS_WW)
nb_subst_NS_SS_all = sum(tab_recap_all_genes_Hydro$Nb_subst_NS_SS)

Nb_total_subst_NS_all = nb_subst_NS_WS_all + nb_subst_NS_SW_all + nb_subst_NS_WW_all + nb_subst_NS_SS_all
Nb_total_subst_S_all = nb_subst_S_WS_all + nb_subst_S_SW_all + nb_subst_S_WW_all + nb_subst_S_SS_all

Nb_totaL_subst_all = Nb_total_subst_NS_all + Nb_total_subst_S_all



###tableau recap pour 15 sp
list_exons_15=read.csv("list_exons_max_15_species.txt", header =F)
colnames(list_exons_15) = c("Exon_ep")
nrow(list_exons_15)

tab_recap_all_genes_Hydro_15 <- merge(tab_recap_all_genes_Hydro , list_exons_15, by = c("Exon_ep"))
library(dplyr)
nrow(distinct(tab_recap_all_genes_Hydro_15, Exon_ep))

nb_subst_S_WS_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_S_WS)
nb_subst_S_SW_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_S_SW)
nb_subst_S_WW_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_S_WW)
nb_subst_S_SS_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_S_SS)

nb_subst_NS_WS_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_NS_WS)
nb_subst_NS_SW_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_NS_SW)
nb_subst_NS_WW_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_NS_WW)
nb_subst_NS_SS_15 = sum(tab_recap_all_genes_Hydro_15$Nb_subst_NS_SS)

Nb_total_subst_NS_15 = nb_subst_NS_WS_15 + nb_subst_NS_SW_15 + nb_subst_NS_WW_15 + nb_subst_NS_SS_15
Nb_total_subst_S_15 = nb_subst_S_WS_15 + nb_subst_S_SW_15 + nb_subst_S_WW_15 + nb_subst_S_SS_15

Nb_totaL_subst_15 = Nb_total_subst_NS_15 + Nb_total_subst_S_15


###tableau recap pour 25 sp
list_exons_25=read.csv("list_exons_max_25_species.txt", header =F)
colnames(list_exons_25) = c("Exon_ep")
nrow(list_exons_25)

tab_recap_all_genes_Hydro_25 <- merge(tab_recap_all_genes_Hydro , list_exons_25, by = c("Exon_ep"))
nrow(distinct(tab_recap_all_genes_Hydro_25, Exon_ep))

nb_subst_S_WS_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_S_WS)
nb_subst_S_SW_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_S_SW)
nb_subst_S_WW_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_S_WW)
nb_subst_S_SS_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_S_SS)

nb_subst_NS_WS_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_NS_WS)
nb_subst_NS_SW_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_NS_SW)
nb_subst_NS_WW_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_NS_WW)
nb_subst_NS_SS_25 = sum(tab_recap_all_genes_Hydro_25$Nb_subst_NS_SS)

Nb_total_subst_NS_25 = nb_subst_NS_WS_25 + nb_subst_NS_SW_25 + nb_subst_NS_WW_25 + nb_subst_NS_SS_25
Nb_total_subst_S_25 = nb_subst_S_WS_25 + nb_subst_S_SW_25 + nb_subst_S_WW_25 + nb_subst_S_SS_25

Nb_totaL_subst_25 = Nb_total_subst_NS_25 + Nb_total_subst_S_25


###tableau recap pour 37 sp
list_exons_37=read.csv("list_exons_max_37_species.txt", header =F)
colnames(list_exons_37) = c("Exon_ep")
nrow(list_exons_37)

tab_recap_all_genes_Hydro_37 <- merge(tab_recap_all_genes_Hydro , list_exons_37, by = c("Exon_ep"))
nrow(distinct(tab_recap_all_genes_Hydro_37, Exon_ep))

nb_subst_S_WS_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_S_WS)
nb_subst_S_SW_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_S_SW)
nb_subst_S_WW_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_S_WW)
nb_subst_S_SS_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_S_SS)

nb_subst_NS_WS_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_NS_WS)
nb_subst_NS_SW_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_NS_SW)
nb_subst_NS_WW_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_NS_WW)
nb_subst_NS_SS_37 = sum(tab_recap_all_genes_Hydro_37$Nb_subst_NS_SS)

Nb_total_subst_NS_37 = nb_subst_NS_WS_37 + nb_subst_NS_SW_37 + nb_subst_NS_WW_37 + nb_subst_NS_SS_37
Nb_total_subst_S_37 = nb_subst_S_WS_37 + nb_subst_S_SW_37 + nb_subst_S_WW_37 + nb_subst_S_SS_37

Nb_totaL_subst_37 = Nb_total_subst_NS_37 + Nb_total_subst_S_37







###plot des subst tot
subst_tot_df <- data.frame(
  nb_esp = c(15, 25, 37, 48),
  total_subst = c(Nb_totaL_subst_15, Nb_totaL_subst_25, Nb_totaL_subst_37, Nb_totaL_subst_all)
)

library(ggplot2)

# Tracer le graphique
ggplot(subst_tot_df, aes(x = factor(nb_esp), y = total_subst)) +
  geom_bar(stat = "identity", fill = "black") +
  xlab("Nombre d'espèces") +
  ylab("Nombre total de substitutions") +
  ggtitle("Nombre total de substitutions selon le nombre d'espèces") +
  theme_minimal()




# Tracer le graphique avec des points
ggplot(subst_tot_df, aes(x = factor(nb_esp), y = total_subst)) +
  geom_point(size = 4, color = "black") +
  geom_line(aes(group = 1), linetype = "dashed", color = "gray40") +
  xlab("Nombre d'espèces") +
  ylab("Nombre total de substitutions") +
  ggtitle("Nombre total de substitutions selon le nombre d'espèces") +
  theme_minimal()



#RATTINI
## table of all data
tab_recap_all_genes_Rattini <- read.csv("tab_recap_all_genes_Rattini.csv", header=FALSE, sep="\t")
colnames(tab_recap_all_genes_Rattini) <- c("Exon_ep","exon_pos", "Lg_seq","GC12","GC3","Br_ep","Br_lg","Br_Asc","Br_Desc1","Br_Desc2","Episode","Nb_subst_S_WS","signif_nb_subst_S_WS","Nb_subst_S_SW","Nb_subst_S_WW","Nb_subst_S_SS","Nb_subst_NS_WS","Nb_subst_NS_SW","Nb_subst_NS_SS","Nb_subst_NS_WW")
###tableau avec des infos sur la branche concernée et tout 


nb_subst_S_WS_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_S_WS)
nb_subst_S_SW_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_S_SW)
nb_subst_S_WW_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_S_WW)
nb_subst_S_SS_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_S_SS)

nb_subst_NS_WS_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_NS_WS)
nb_subst_NS_SW_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_NS_SW)
nb_subst_NS_WW_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_NS_WW)
nb_subst_NS_SS_Rattini = sum(tab_recap_all_genes_Rattini$Nb_subst_NS_SS)

Nb_total_subst_NS_Rattini = nb_subst_NS_WS_Rattini + nb_subst_NS_SW_Rattini + nb_subst_NS_WW_Rattini + nb_subst_NS_SS_Rattini
Nb_total_subst_S_Rattini = nb_subst_S_WS_Rattini + nb_subst_S_SW_Rattini + nb_subst_S_WW_Rattini + nb_subst_S_SS_Rattini

Nb_totaL_subst_Rattini = Nb_total_subst_NS_Rattini + Nb_total_subst_S_Rattini







#CÉTACÉS

## table of all data
tab_recap_all_genes_Cetacea <- read.csv("tab_recap_all_genes_Cetacea.csv", header=FALSE, sep="\t")

colnames(tab_recap_all_genes_Cetacea) <- c("Exon_ep", "Lg_seq","GC12","GC3","Br_ep","Br_lg","Br_Asc","Br_Desc1","Br_Desc2","Episode","Nb_subst_S_WS","signif_nb_subst_S_WS","Nb_subst_S_SW","Nb_subst_S_WW","Nb_subst_S_SS","Nb_subst_NS_WS","Nb_subst_NS_SW","Nb_subst_NS_SS","Nb_subst_NS_WW")
###tableau avec des infos sur la branche concernée et tout 


nb_subst_S_WS_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_S_WS)
nb_subst_S_SW_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_S_SW)
nb_subst_S_WW_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_S_WW)
nb_subst_S_SS_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_S_SS)

nb_subst_NS_WS_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_NS_WS)
nb_subst_NS_SW_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_NS_SW)
nb_subst_NS_WW_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_NS_WW)
nb_subst_NS_SS_Cetacea = sum(tab_recap_all_genes_Cetacea$Nb_subst_NS_SS)

Nb_total_subst_NS_Cetacea = nb_subst_NS_WS_Cetacea + nb_subst_NS_SW_Cetacea + nb_subst_NS_WW_Cetacea + nb_subst_NS_SS_Cetacea
Nb_total_subst_S_Cetacea = nb_subst_S_WS_Cetacea + nb_subst_S_SW_Cetacea + nb_subst_S_WW_Cetacea + nb_subst_S_SS_Cetacea

Nb_totaL_subst_Cetacea = Nb_total_subst_NS_Cetacea + Nb_total_subst_S_Cetacea


#CÉTACÉS_25

## table of all data
tab_recap_all_genes_Cetacea_25 <- read.csv("Cetacea_25/tab_recap_all_genes_Cetacea.csv", header=FALSE, sep="\t")

colnames(tab_recap_all_genes_Cetacea_25) <- c("Exon_ep", "Lg_seq","GC12","GC3","Br_ep","Br_lg","Br_Asc","Br_Desc1","Br_Desc2","Episode","Nb_subst_S_WS","signif_nb_subst_S_WS","Nb_subst_S_SW","Nb_subst_S_WW","Nb_subst_S_SS","Nb_subst_NS_WS","Nb_subst_NS_SW","Nb_subst_NS_SS","Nb_subst_NS_WW")
###tableau avec des infos sur la branche concernée et tout 


nb_subst_S_WS_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_S_WS)
nb_subst_S_SW_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_S_SW)
nb_subst_S_WW_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_S_WW)
nb_subst_S_SS_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_S_SS)

nb_subst_NS_WS_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_NS_WS)
nb_subst_NS_SW_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_NS_SW)
nb_subst_NS_WW_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_NS_WW)
nb_subst_NS_SS_Cetacea_25 = sum(tab_recap_all_genes_Cetacea_25$Nb_subst_NS_SS)

Nb_total_subst_NS_Cetacea_25 = nb_subst_NS_WS_Cetacea_25 + nb_subst_NS_SW_Cetacea_25 + nb_subst_NS_WW_Cetacea_25 + nb_subst_NS_SS_Cetacea_25
Nb_total_subst_S_Cetacea_25 = nb_subst_S_WS_Cetacea_25 + nb_subst_S_SW_Cetacea_25 + nb_subst_S_WW_Cetacea_25 + nb_subst_S_SS_Cetacea_25

Nb_totaL_subst_Cetacea_25 = Nb_total_subst_NS_Cetacea_25 + Nb_total_subst_S_Cetacea_25


###MURINI

## table of all data
tab_recap_all_genes_Murini <- read.csv("tab_recap_all_genes_Murini.csv", header=FALSE, sep="\t")
colnames(tab_recap_all_genes_Murini) = c("Exon_ep","exon_pos", "Dn/Ds", "Lg_seq","GC12","GC3","Br_ep","Br_lg","Br_Asc","Br_Desc1","Br_Desc2","Episode","EpisodeSW","EpisodeWSSW","Episode_wtI","Nb_subst_S_WS","signif_nb_subst_S_WS","Nb_subst_S_SW","Nb_subst_S_WW","Nb_subst_S_SS","Nb_subst_NS_WS","Nb_subst_NS_SW","Nb_subst_NS_SS","Nb_subst_NS_WW")
###tableau avec des infos sur la branche concernée et tout 



nb_subst_S_WS_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_S_WS)
nb_subst_S_SW_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_S_SW)
nb_subst_S_WW_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_S_WW)
nb_subst_S_SS_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_S_SS)

nb_subst_NS_WS_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_NS_WS)
nb_subst_NS_SW_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_NS_SW)
nb_subst_NS_WW_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_NS_WW)
nb_subst_NS_SS_Murini = sum(tab_recap_all_genes_Murini$Nb_subst_NS_SS)

Nb_total_subst_NS_Murini = nb_subst_NS_WS_Murini + nb_subst_NS_SW_Murini + nb_subst_NS_WW_Murini + nb_subst_NS_SS_Murini
Nb_total_subst_S_Murini = nb_subst_S_WS_Murini + nb_subst_S_SW_Murini + nb_subst_S_WW_Murini + nb_subst_S_SS_Murini

Nb_totaL_subst_Murini = Nb_total_subst_NS_Murini + Nb_total_subst_S_Murini







# Tracer le graphique avec des points

# Ajouter la ligne pour Rattini
subst_tot_CR <- rbind(subst_tot_df, data.frame(nb_esp = "Rattini", total_subst = Nb_totaL_subst_Rattini))
subst_tot_CR <- rbind(subst_tot_CR, data.frame(nb_esp = "Cetacea", total_subst = Nb_totaL_subst_Cetacea))
subst_tot_CR <- rbind(subst_tot_CR, data.frame(nb_esp = "Murini", total_subst = Nb_totaL_subst_Murini))
subst_tot_CR <- rbind(subst_tot_CR, data.frame(nb_esp = "Cetacea_25_sp", total_subst = Nb_totaL_subst_Cetacea_25))

# Exemple pour renommer une colonne
colnames(subst_tot_CR)[colnames(subst_tot_CR) == "nb_esp"] <- "Groupe"


# Exemple pour changer le nom des lignes
rownames(subst_tot_CR) <- c("15 sp", "25 sp", "37 sp", "48 sp", "Rattini", "Cetacea","Cetacea 25 sp", "Murini")

# Créer une colonne 'Groupe' correspondant au nom des lignes
subst_tot_CR$Groupe <- rownames(subst_tot_CR)

# Transformer 'Groupe' en facteur avec ordre croissant basé sur 'total_subst'
subst_tot_CR$Groupe <- factor(subst_tot_CR$Groupe, levels = subst_tot_CR$Groupe[order(subst_tot_CR$total_subst)])

# Créer une colonne 'groupe' pour l'affichage des couleurs
subst_tot_CR$groupe <- ifelse(subst_tot_CR$Groupe == "Rattini", "Rattini",
                        ifelse(subst_tot_CR$Groupe == "Cetacea", "Cetacea",
                        ifelse(subst_tot_CR$Groupe == "Cetacea 25 sp", "Cetacea 25 sp",
                        ifelse(subst_tot_CR$Groupe == "Murini", "Murini", "Hydromyini"))))

# Tracer le graphique
ggplot(subst_tot_CR, aes(x = Groupe, y = total_subst, color = groupe)) +
  geom_point(size = 3) +
  scale_color_manual(values = c("Rattini" = "red", "Cetacea" = "blue", "Cetacea 25 sp" ="pink", "Murini" = "green", "Hydromyini" = "black")) +
  xlab("Groupe") +
  ylab("Nombre total de substitutions") +
 # ggtitle("Nombre total de substitutions selon le groupe") +
  theme_minimal()

# Tracer l'histogramme du nombre total de substitutions par Groupe
plot_subst=ggplot(subst_tot_CR, aes(x = Groupe, y = total_subst, fill = groupe)) +
  geom_bar(stat = "identity", color = "black") +
  scale_fill_manual(values = c(
    "Rattini" = "#ee9b00",
    "Cetacea" = "#005f73",
    "Cetacea 25 sp" = "#94d2bd",
    "Murini" = "#e9d8a6",
    "Hydromyini" = "#9b2226"
  )) +
  xlab("Groupe") +
  ylab("Nombre total de substitutions") +
  # ggtitle("Nombre total de substitutions selon le groupe") +
  theme_minimal()

plot_subst

ggsave("plot_subst.png", plot_subst)


```

###Z score selon le nb d'espèces
```{r}
##15 sp
df_15_WS = subset(df_points_15, df_points_15$variable == "WS")
df_15_WS <- df_15_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_SW = subset(df_points_15, df_points_15$variable == "SW")
df_15_SW <- df_15_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_SSWW = subset(df_points_15, df_points_15$variable == "SSWW")
df_15_SSWW <- df_15_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_total = subset(df_points_15, df_points_15$variable == "total")
df_15_total <- df_15_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

#25sp

df_25_WS = subset(df_points_25, df_points_25$variable == "WS")
df_25_WS <- df_25_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_SW = subset(df_points_25, df_points_25$variable == "SW")
df_25_SW <- df_25_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_SSWW = subset(df_points_25, df_points_25$variable == "SSWW")
df_25_SSWW <- df_25_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_total = subset(df_points_25, df_points_25$variable == "total")
df_25_total <- df_25_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

#37
df_37_WS = subset(df_points_37, df_points_37$variable == "WS")
df_37_WS <- df_37_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_SW = subset(df_points_37, df_points_37$variable == "SW")
df_37_SW <- df_37_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_SSWW = subset(df_points_37, df_points_37$variable == "SSWW")
df_37_SSWW <- df_37_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_total = subset(df_points_37, df_points_37$variable == "total")
df_37_total <- df_37_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

##48sp
df_48_WS = subset(df_points_48, df_points_48$variable == "WS")
df_48_WS <- df_48_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_SW = subset(df_points_48, df_points_48$variable == "SW")
df_48_SW <- df_48_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_SSWW = subset(df_points_48, df_points_48$variable == "SSWW")
df_48_SSWW <- df_48_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_total = subset(df_points_48, df_points_48$variable == "total")
df_48_total <- df_48_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  



###Murini
df_Murini_WS = subset(df_points_Murini, df_points_Murini$variable == "WS")
df_Murini_WS <- df_Murini_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_SW = subset(df_points_Murini, df_points_Murini$variable == "SW")
df_Murini_SW <- df_Murini_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_SSWW = subset(df_points_Murini, df_points_Murini$variable == "SSWW")
df_Murini_SSWW <- df_Murini_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_total = subset(df_points_Murini, df_points_Murini$variable == "total")
df_Murini_total <- df_Murini_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


###Rattini

df_Rattini_WS = subset(df_points_Rattini, df_points_Rattini$variable == "WS")
df_Rattini_WS <- df_Rattini_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_SW = subset(df_points_Rattini, df_points_Rattini$variable == "SW")
df_Rattini_SW <- df_Rattini_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_SSWW = subset(df_points_Rattini, df_points_Rattini$variable == "SSWW")
df_Rattini_SSWW <- df_Rattini_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_total = subset(df_points_Rattini, df_points_Rattini$variable == "total")
df_Rattini_total <- df_Rattini_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

####Cétacea

df_Cetacea_WS = subset(df_points_Cetacea, df_points_Cetacea$variable == "WS")
df_Cetacea_WS <- df_Cetacea_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_SW = subset(df_points_Cetacea, df_points_Cetacea$variable == "SW")
df_Cetacea_SW <- df_Cetacea_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_SSWW = subset(df_points_Cetacea, df_points_Cetacea$variable == "SSWW")
df_Cetacea_SSWW <- df_Cetacea_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_total = subset(df_points_Cetacea, df_points_Cetacea$variable == "total")
df_Cetacea_total <- df_Cetacea_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


####Cétacea 25 sp

df_Cetacea_25_WS = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "WS")
df_Cetacea_25_WS <- df_Cetacea_25_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_SW = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "SW")
df_Cetacea_25_SW <- df_Cetacea_25_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_SSWW = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "SSWW")
df_Cetacea_25_SSWW <- df_Cetacea_25_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_total = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "total")
df_Cetacea_25_total <- df_Cetacea_25_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

####merge les tableaux 



diff_WS <- rbind(
  df_15_WS %>% mutate(nb_esp = "15"),
  df_25_WS %>% mutate(nb_esp = "25"),
  df_37_WS %>% mutate(nb_esp = "37"),
  df_48_WS %>% mutate(nb_esp = "48"),
  df_Murini_WS %>% mutate(nb_esp = "Murini"),
  df_Rattini_WS %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_WS %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_WS %>% mutate(nb_esp = "Cetacea_25")
)


diff_SW <- rbind(
  df_15_SW %>% mutate(nb_esp = "15"),
  df_25_SW %>% mutate(nb_esp = "25"),
  df_37_SW %>% mutate(nb_esp = "37"),
  df_48_SW %>% mutate(nb_esp = "48"),
  df_Murini_SW %>% mutate(nb_esp = "Murini"),
  df_Rattini_SW %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_SW %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_SW %>% mutate(nb_esp = "Cetacea_25")
)


diff_SSWW <- rbind(
  df_15_SSWW %>% mutate(nb_esp = "15"),
  df_25_SSWW %>% mutate(nb_esp = "25"),
  df_37_SSWW %>% mutate(nb_esp = "37"),
  df_48_SSWW %>% mutate(nb_esp = "48"),
  df_Murini_SSWW %>% mutate(nb_esp = "Murini"),
  df_Rattini_SSWW %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_SSWW %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_SSWW %>% mutate(nb_esp = "Cetacea_25")
)




diff_total <- rbind(
  df_15_total %>% mutate(nb_esp = "15"),
  df_25_total %>% mutate(nb_esp = "25"),
  df_37_total %>% mutate(nb_esp = "37"),
  df_48_total %>% mutate(nb_esp = "48"),
  df_Murini_total %>% mutate(nb_esp = "Murini"),
  df_Rattini_total %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_total %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_total %>% mutate(nb_esp = "Cetacea_25")
)


ggplot(diff_WS, aes(x = factor(nb_esp), y = diff)) +
  geom_point(size = 3, color = "red") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-score : NS - no_NS pour WS selon le nombre d'espèces") +
  theme_minimal()

ggplot(diff_SW, aes(x = factor(nb_esp), y = diff)) +
  geom_point(size = 3, color = "cornflowerblue") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-score : NS - no_NS pour SW selon le nombre d'espèces") +
  theme_minimal()

ggplot(diff_SSWW, aes(x = factor(nb_esp), y = diff)) +
  geom_point(size = 3, color = "darkseagreen") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-score : NS - no_NS pour SSWW selon le nombre d'espèces") +
  theme_minimal()

ggplot(diff_total, aes(x = factor(nb_esp), y = diff)) +
  geom_point(size = 3, color = "plum3") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("sélection compensatoire") +
  theme_minimal()
```

###Z score selon le nb de substitutions 
```{r}

###plot des subst tot
subst_tot_df <- data.frame(
  Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp"),
  total_subst = c(Nb_totaL_subst_15, Nb_totaL_subst_25, Nb_totaL_subst_37, Nb_totaL_subst_all, Nb_totaL_subst_Murini, Nb_totaL_subst_Rattini, Nb_totaL_subst_Cetacea, Nb_totaL_subst_Cetacea_25)
)

###Rajouter une colonne
df_points_15$nb_esp <- 15
df_points_25$nb_esp <- 25
df_points_37$nb_esp <- 37
df_points_48$nb_esp <- 48

# Fusionner tous les data frames

df_all <- rbind(df_points_15, df_points_25, df_points_37, df_points_48)
##15 sp
df_15_WS = subset(df_points_15, df_points_15$variable == "WS")
df_15_WS <- df_15_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_SW = subset(df_points_15, df_points_15$variable == "SW")
df_15_SW <- df_15_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_SSWW = subset(df_points_15, df_points_15$variable == "SSWW")
df_15_SSWW <- df_15_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_total = subset(df_points_15, df_points_15$variable == "total")
df_15_total <- df_15_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

#25sp

df_25_WS = subset(df_points_25, df_points_25$variable == "WS")
df_25_WS <- df_25_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_SW = subset(df_points_25, df_points_25$variable == "SW")
df_25_SW <- df_25_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_SSWW = subset(df_points_25, df_points_25$variable == "SSWW")
df_25_SSWW <- df_25_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_total = subset(df_points_25, df_points_25$variable == "total")
df_25_total <- df_25_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

#37
df_37_WS = subset(df_points_37, df_points_37$variable == "WS")
df_37_WS <- df_37_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_SW = subset(df_points_37, df_points_37$variable == "SW")
df_37_SW <- df_37_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_SSWW = subset(df_points_37, df_points_37$variable == "SSWW")
df_37_SSWW <- df_37_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_total = subset(df_points_37, df_points_37$variable == "total")
df_37_total <- df_37_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

##48sp
df_48_WS = subset(df_points_48, df_points_48$variable == "WS")
df_48_WS <- df_48_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_SW = subset(df_points_48, df_points_48$variable == "SW")
df_48_SW <- df_48_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_SSWW = subset(df_points_48, df_points_48$variable == "SSWW")
df_48_SSWW <- df_48_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_total = subset(df_points_48, df_points_48$variable == "total")
df_48_total <- df_48_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


###Murini
df_Murini_WS = subset(df_points_Murini, df_points_Murini$variable == "WS")
df_Murini_WS <- df_Murini_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_SW = subset(df_points_Murini, df_points_Murini$variable == "SW")
df_Murini_SW <- df_Murini_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_SSWW = subset(df_points_Murini, df_points_Murini$variable == "SSWW")
df_Murini_SSWW <- df_Murini_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_total = subset(df_points_Murini, df_points_Murini$variable == "total")
df_Murini_total <- df_Murini_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


###Rattini

df_Rattini_WS = subset(df_points_Rattini, df_points_Rattini$variable == "WS")
df_Rattini_WS <- df_Rattini_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_SW = subset(df_points_Rattini, df_points_Rattini$variable == "SW")
df_Rattini_SW <- df_Rattini_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_SSWW = subset(df_points_Rattini, df_points_Rattini$variable == "SSWW")
df_Rattini_SSWW <- df_Rattini_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_total = subset(df_points_Rattini, df_points_Rattini$variable == "total")
df_Rattini_total <- df_Rattini_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

####Cétacea

df_Cetacea_WS = subset(df_points_Cetacea, df_points_Cetacea$variable == "WS")
df_Cetacea_WS <- df_Cetacea_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_SW = subset(df_points_Cetacea, df_points_Cetacea$variable == "SW")
df_Cetacea_SW <- df_Cetacea_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_SSWW = subset(df_points_Cetacea, df_points_Cetacea$variable == "SSWW")
df_Cetacea_SSWW <- df_Cetacea_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_total = subset(df_points_Cetacea, df_points_Cetacea$variable == "total")
df_Cetacea_total <- df_Cetacea_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


####Cétacea 25 sp

df_Cetacea_25_WS = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "WS")
df_Cetacea_25_WS <- df_Cetacea_25_WS %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_SW = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "SW")
df_Cetacea_25_SW <- df_Cetacea_25_SW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_SSWW = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "SSWW")
df_Cetacea_25_SSWW <- df_Cetacea_25_SSWW %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_total = subset(df_points_Cetacea_25, df_points_Cetacea_25$variable == "total")
df_Cetacea_25_total <- df_Cetacea_25_total %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

####merge les tableaux 



diff_WS <- rbind(
  df_15_WS %>% mutate(nb_esp = "15"),
  df_25_WS %>% mutate(nb_esp = "25"),
  df_37_WS %>% mutate(nb_esp = "37"),
  df_48_WS %>% mutate(nb_esp = "48"),
  df_Murini_WS %>% mutate(nb_esp = "Murini"),
  df_Rattini_WS %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_WS %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_WS %>% mutate(nb_esp = "Cetacea_25")
)


diff_SW <- rbind(
  df_15_SW %>% mutate(nb_esp = "15"),
  df_25_SW %>% mutate(nb_esp = "25"),
  df_37_SW %>% mutate(nb_esp = "37"),
  df_48_SW %>% mutate(nb_esp = "48"),
  df_Murini_SW %>% mutate(nb_esp = "Murini"),
  df_Rattini_SW %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_SW %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_SW %>% mutate(nb_esp = "Cetacea_25")
)


diff_SSWW <- rbind(
  df_15_SSWW %>% mutate(nb_esp = "15"),
  df_25_SSWW %>% mutate(nb_esp = "25"),
  df_37_SSWW %>% mutate(nb_esp = "37"),
  df_48_SSWW %>% mutate(nb_esp = "48"),
  df_Murini_SSWW %>% mutate(nb_esp = "Murini"),
  df_Rattini_SSWW %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_SSWW %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_SSWW %>% mutate(nb_esp = "Cetacea_25")
)




diff_total <- rbind(
  df_15_total %>% mutate(nb_esp = "15"),
  df_25_total %>% mutate(nb_esp = "25"),
  df_37_total %>% mutate(nb_esp = "37"),
  df_48_total %>% mutate(nb_esp = "48"),
  df_Murini_total %>% mutate(nb_esp = "Murini"),
  df_Rattini_total %>% mutate(nb_esp = "Rattini"),
  df_Cetacea_total %>% mutate(nb_esp = "Cetacea"),
  df_Cetacea_25_total %>% mutate(nb_esp = "Cetacea_25")
)



###changer le nom des lignes et de la colonne
diff_WS_all <- diff_WS[, c(1, 3:5)]  
diff_WS_all$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")

diff_SW_all <- diff_SW[, c(1, 3:5)]  
diff_SW_all$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")

diff_SSWW_all <- diff_SSWW[, c(1, 3:5)]  
diff_SSWW_all$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")

diff_total_all <- diff_total[, c(1, 3:5)]  
diff_total_all$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")


data_subst_WS = merge(diff_WS_all, subst_tot_df, by = c("Groupe"))
data_subst_SW = merge(diff_SW_all, subst_tot_df, by = c("Groupe"))
data_subst_SSWW = merge(diff_SSWW_all, subst_tot_df, by = c("Groupe"))
data_subst_total = merge(diff_total_all, subst_tot_df, by = c("Groupe"))



ggplot(data_subst_WS, aes(x = factor(total_subst), y = diff)) +
  geom_point(size = 3, color = "red") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-score : NS - no_NS pour WS selon le nombre d'espèces") +
  theme_minimal()

ggplot(data_subst_SW, aes(x = factor(total_subst), y = diff)) +
  geom_point(size = 3, color = "cornflowerblue") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-score : NS - no_NS pour SW selon le nombre d'espèces") +
  theme_minimal()

ggplot(data_subst_SSWW, aes(x = factor(total_subst), y = diff)) +
  geom_point(size = 3, color = "darkseagreen") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-score : NS - no_NS pour SSWW selon le nombre d'espèces") +
  theme_minimal()

ggplot(data_subst_total, aes(x = factor(total_subst), y = diff)) +
  geom_point(size = 3, color = "plum3") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("sélection compensatoire") +
  xlab("Nombre d'espèces") +
  ggtitle("sélection compensatoire") +
  theme_minimal()


# Définir une palette de couleurs personnalisée
couleurs_Groupes <- c(
  "15_sp_Hydro" = "#1b9e77",
  "25_sp_Hydro" = "#24D26D",
  "37_sp_Hydro" = "darkgreen",
  "48_sp_Hydro" = "#CDEAC0",
  "Murini"       = "#d95f02",
  "Rattini"      = "#7570b3",
  "Cetacea"      = "#e7298a",
  "Cetacea_25_sp" = "pink"
)

plot_subst_WS_lg_inf_0.006=ggplot(data_subst_WS, aes(x = (total_subst), y = diff, color = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(method = "lm", se = T, color = "black", size = 0.5) +
  scale_color_manual(values = couleurs_Groupes) +
  ylab("sélection compensatoire") +
  xlab("Nombre de substitutions") +
  ggtitle("Z-score : NS - no_NS pour WS selon le nombre d'espèces") +
  theme_minimal()

plot_subst_WS_lg_inf_0.006

plot_subst_SW_lg_inf_0.006=ggplot(data_subst_SW, aes(x = (total_subst), y = diff, color = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(method = "lm", se = T, color = "black", size = 0.5) +
  scale_color_manual(values = couleurs_Groupes) +
  ylab("sélection compensatoire") +
  xlab("Nombre de substitutions") +
  ggtitle("Z-score : NS - no_NS pour SW selon le nombre d'espèces") +
  theme_minimal()
plot_subst_SW_lg_inf_0.006

plot_subst_SSWW_lg_inf_0.006=ggplot(data_subst_SSWW, aes(x = (total_subst), y = diff, color = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(method = "lm", se = T, color = "black", size = 0.5) +
  scale_color_manual(values = couleurs_Groupes) +
  ylab("sélection compensatoire") +
  xlab("Nombre de substitutions") +
  ggtitle("Z-score : NS - no_NS pour SSWW selon le nombre d'espèces") +
  theme_minimal()

plot_subst_SSWW_lg_inf_0.006

plot_subst_total_lg_inf_0.006=ggplot(data_subst_total, aes(x = (total_subst), y = diff, color = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(method = "lm", se = T, color = "black", size = 0.5) +
  scale_color_manual(values = couleurs_Groupes) +
  ylab("sélection compensatoire") +
  xlab("Nombre de substitutions") +
  ggtitle("sélection compensatoire") +
  theme_minimal()

plot_subst_total_lg_inf_0.006

```










####Pas de longueurs de branches imposées 




##Sous écha = 15 sp###
```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
#tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)
```

```{r}
hist(tab_ep_ponctual$post1,
     main = "Histogramme post 1",
     xlab = "post 1",
     ylab = "Fréquence",
     col = "lightblue",
     border = "white",
     breaks = 350)
```




```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_15sp/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)

#moy_NS_WS_obs = mean(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)
#moy_NS_WS_att <- as.data.frame(tab_random_NS_WS_postepNS[, 18:1017])
#mean(moy_NS_WS_att)


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_15sp/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_15sp/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_15sp/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_15sp/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_15sp/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_15sp/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_15sp/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual










nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_15_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_15_all <- melt(df_Z_scores_15_all, varnames = c("Condition", "Transition"))
df_points_15_all$Condition <- rep(rownames(df_Z_scores_15_all), times = ncol(df_Z_scores_15_all))

plot_15_sp_all = ggplot(df_points_15_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_15_sp_all


```








































##Sous écha = 25 sp
```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_25sp/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)

#moy_NS_WS_obs = mean(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)
#moy_NS_WS_att <- as.data.frame(tab_random_NS_WS_postepNS[, 18:1017])
#mean(moy_NS_WS_att)


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_25sp/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_25sp/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_25sp/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_25sp/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_25sp/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_25sp/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_25sp/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual













nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_25_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_25_all <- melt(df_Z_scores_25_all, varnames = c("Condition", "Transition"))
df_points_25_all$Condition <- rep(rownames(df_Z_scores_25_all), times = ncol(df_Z_scores_25_all))

plot_25_sp_all=ggplot(df_points_25_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_25_sp_all


```















### Sous échan = 37 sp

```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_37sp/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit


nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)

#moy_NS_WS_obs = mean(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)
#moy_NS_WS_att <- as.data.frame(tab_random_NS_WS_postepNS[, 18:1017])
#mean(moy_NS_WS_att)


## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_37sp/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_37sp/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_37sp/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_37sp/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_37sp/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_37sp/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_37sp/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep","Gene","true_type", "post1", "post2", "post3", "GoF")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual













nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_37_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_37_all <- melt(df_Z_scores_37_all, varnames = c("Condition", "Transition"))
df_points_37_all$Condition <- rep(rownames(df_Z_scores_37_all), times = ncol(df_Z_scores_37_all))

plot_37_sp_all=ggplot(df_points_37_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_37_sp_all


```














## Sous échan = 48sp

```{r}
tab_tri_ep <- read.csv("posterior_episodes_optimum_hydromyini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_hydromyini.txt", header=F)
exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("max_48sp/tab_NS_WS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("max_48sp/tab_NS_SW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("max_48sp/tab_NS_SS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("max_48sp/tab_NS_WW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

	## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("max_48sp/tab_NS_WS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("max_48sp/tab_NS_SW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit


# load table
tab_random_NS_SS_postepnoNS <- read.csv("max_48sp/tab_NS_SS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("max_48sp/tab_NS_WW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

	## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual












nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_48_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_48_all <- melt(df_Z_scores_48_all, varnames = c("Condition", "Transition"))
df_points_48_all$Condition <- rep(rownames(df_Z_scores_48_all), times = ncol(df_Z_scores_48_all))

plot_48_sp_all=ggplot(df_points_48_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_48_sp_all
```




```{r}

plot_all_Z <- ggarrange(plot_15_sp_all, plot_25_sp_all, plot_37_sp_all, plot_48_sp_all, nrow=2, ncol = 2)
plot_all_Z

```









##Z score NS vs no NS
```{r}

df_points_15_all$nb_esp <- 15
df_points_25_all$nb_esp <- 25
df_points_37_all$nb_esp <- 37
df_points_48_all$nb_esp <- 48

# Fusionner tous les data frames en les empilant verticalement
df_all_all_lg <- rbind(df_points_15_all, df_points_25_all, df_points_37_all, df_points_48_all)
# Plot final
ggplot(df_all_all_lg[df_all_all_lg$variable == "WS", ], aes(x = factor(nb_esp), y = value, fill = Condition)) +
  geom_col(position = "dodge") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour WS selon le nombre d'espèces") +
  scale_fill_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

ggplot(df_all_all_lg[df_all_all_lg$variable == "SW", ], aes(x = factor(nb_esp), y = value, fill = Condition)) +
  geom_col(position = "dodge") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour WS selon le nombre d'espèces") +
  scale_fill_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

ggplot(df_all_all_lg[df_all_all_lg$variable == "SSWW", ], aes(x = factor(nb_esp), y = value, fill = Condition)) +
  geom_col(position = "dodge") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour WS selon le nombre d'espèces") +
  scale_fill_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

ggplot(df_all_all_lg[df_all_all_lg$variable == "total", ], aes(x = factor(nb_esp), y = value, fill = Condition)) +
  geom_col(position = "dodge") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour WS selon le nombre d'espèces") +
  scale_fill_manual(values = c("steelblue", "tomato")) +
  theme_minimal()



ggplot(df_all_all_lg[df_all_all_lg$variable == "WS", ], aes(x = factor(nb_esp), y = value, color = Condition)) +
  geom_point(position = position_dodge(width = 0.5), size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour WS selon le nombre d'espèces") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()



ggplot(df_all_all_lg[df_all_all_lg$variable == "SW", ], aes(x = factor(nb_esp), y = value, color = Condition)) +
  geom_point(position = position_dodge(width = 0.5), size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour SW selon le nombre d'espèces") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

ggplot(df_all_all_lg[df_all_all_lg$variable == "SSWW", ], aes(x = factor(nb_esp), y = value, color = Condition)) +
  geom_point(position = position_dodge(width = 0.5), size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour SSWW selon le nombre d'espèces") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()
ggplot(df_all_all_lg[df_all_all_lg$variable == "total", ], aes(x = factor(nb_esp), y = value, color = Condition)) +
  geom_point(position = position_dodge(width = 0.5), size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Nombre d'espèces") +
  ggtitle("Z-scores NS vs no_NS pour total selon le nombre d'espèces") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()



```



####Murini

```{r}

tab_tri_ep <- read.csv("posterior_episodes_optimum_murini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("paralog_exons_murini.txt", header=F)
#exons_aln_error <- read.table("List_exons_error_aln.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
#tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)

	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

hist(tab_ep_ponctual$post1,
     main = "Histogramme post 1",
     xlab = "post 1",
     ylab = "Fréquence",
     col = "lightblue",
     border = "white",
     breaks = 350)






## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

	## WS

# load table
tab_random_NS_WS_postepNS <- read.csv("tab_all_genes_murini/tab_NS_WS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("tab_all_genes_murini/tab_NS_SW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit

## SS

# load table
tab_random_NS_SS_postepNS <- read.csv("tab_all_genes_murini/tab_NS_SS_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepNS <- read.csv("tab_all_genes_murini/tab_NS_WW_after_epNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit

## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit

## plots
# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual
# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS

# NS after episodes with no NS-WS #

	## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_WS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit

## SW

# load table
tab_random_NS_SW_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_SW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit


# load table
tab_random_NS_SS_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_SS_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")

## WW

# load table
tab_random_NS_WW_postepnoNS <- read.csv("tab_all_genes_murini/tab_NS_WW_after_epnoNS_with_random.csv", header=F, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 18:1017]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit

	## total NS = WS+SW+SSWW
# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit

	## plots
# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual












nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Murini_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Murini_all <- melt(df_Z_scores_Murini_all, varnames = c("Condition", "Transition"))
df_points_Murini_all$Condition <- rep(rownames(df_Z_scores_Murini_all), times = ncol(df_Z_scores_Murini_all))

plot_Murini_all = ggplot(df_points_Murini_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Murini_all







```


###Rattini

```{r}




### compensation analyses ###

## Episode classification


tab_tri_ep <- read.csv("Rattini/posterior_optimum_Rattini.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("Rattini/paralog_exons_Rattini.txt", header=F)
exons_aln_error <- read.table("Rattini/List_exons_error_aln_Rattini.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.60)
	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

###récupère les tableaux des épisodes ponctuels, deux branches et tous les exons




## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

## WS


# load table
tab_random_NS_WS_postepNS <- read.csv("Rattini/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
#j'ai changé le nom des colonnes, j'ai enlevé le dn/ds
colnames(tab_random_NS_WS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

#colnames(tab_ep_ponctual)
#colnames(tab_random_NS_WS_postepNS)

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]
###j'ai mis un = pour dire qu'on garde quand on a un épisode 
#tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit



## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("Rattini/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_SW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]
#tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)


# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit
	


#SS

# load table
tab_random_NS_SS_postepNS <- read.csv("Rattini/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

#colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br_ep', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'))
colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")


# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]
#tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 17:1016]))

colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")
	


#WW

# load table
tab_random_NS_WW_postepNS <- read.csv("Rattini/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_WW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")



# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]
#tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit



## total NS = WS+SW+SSWW

# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit


	## plots


# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual

# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS






# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("Rattini/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit



## SW




# load table
tab_random_NS_SW_postepnoNS <- read.csv("Rattini/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit



## SS



# load table
tab_random_NS_SS_postepnoNS <- read.csv("Rattini/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")



## WW




# load table
tab_random_NS_WW_postepnoNS <- read.csv("Rattini/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit


## total NS = WS+SW+SSWW


# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit


# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual







nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Rattini_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Rattini_all <- melt(df_Z_scores_Rattini_all, varnames = c("Condition", "Transition"))
df_points_Rattini_all$Condition <- rep(rownames(df_Z_scores_Rattini_all), times = ncol(df_Z_scores_Rattini_all))

plot_Rattini_all = ggplot(df_points_Rattini_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Rattini_all

```



#Cetacea

```{r}
### compensation analyses ###

## Episode classification


tab_tri_ep <- read.csv("Cetacea/posterior_optimum_Cetacea.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
#exons_paralog <- read.csv("paralog_exons.txt", header=F)
exons_aln_error <- read.table("Cetacea/List_exons_error_aln.csv", header=T)
#tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.6)
	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

#tab_ep_ponctual_2 = merge(tab_ep_ponctual, tab, by = c("Exon_ep", "Br_ep"))
#mean(tab_ep_ponctual_2$Br_lg)

#tab_ep_mean = merge(tab_tri_ep, tab, by = c("Exon_ep", "Br_ep"))
#mean(tab_ep_mean$Br_lg)
###récupère les tableaux des épisodes ponctuels, deux branches et tous les exons

#mean(tab$Br_lg)
## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

## WS


# load table
tab_random_NS_WS_postepNS <- read.csv("Cetacea/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
#j'ai changé le nom des colonnes, j'ai enlevé le dn/ds
colnames(tab_random_NS_WS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

#colnames(tab_ep_ponctual)
#colnames(tab_random_NS_WS_postepNS)

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]
###j'ai mis un = pour dire qu'on garde quand on a un épisode 
#tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg > 0.004)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit



## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("Cetacea/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_SW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]
#tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)


# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit
	


#SS

# load table
tab_random_NS_SS_postepNS <- read.csv("Cetacea/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

#colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br_ep', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'))
colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")


# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]
#tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 17:1016]))

colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")
	


#WW

# load table
tab_random_NS_WW_postepNS <- read.csv("Cetacea/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_WW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")



# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]
#tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit




## total NS = WS+SW+SSWW

# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit


	## plots


# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual

# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS








# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("Cetacea/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit





## SW




# load table
tab_random_NS_SW_postepnoNS <- read.csv("Cetacea/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit



## SS



# load table
tab_random_NS_SS_postepnoNS <- read.csv("Cetacea/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")



## WW




# load table
tab_random_NS_WW_postepnoNS <- read.csv("Cetacea/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit


## total NS = WS+SW+SSWW


# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit



## plots


# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual








nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Cetacea_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Cetacea_all <- melt(df_Z_scores_Cetacea_all, varnames = c("Condition", "Transition"))
df_points_Cetacea_all$Condition <- rep(rownames(df_Z_scores_Cetacea_all), times = ncol(df_Z_scores_Cetacea_all))

plot_Cetacea_all = ggplot(df_points_Cetacea_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Cetacea_all


```



###Cetacea_25_sp

```{r}
### compensation analyses ###

## Episode classification


tab_tri_ep <- read.csv("Cetacea_25/posterior_optimum_Cetacea.csv", header=T, sep=",")
colnames(tab_tri_ep) <- c("Gene", "Exon_ep", "Br_ep", "true_type", "post1", "post2", "post3", "GoF")
	# retrieve paralog exonx and aln with potential error
exons_paralog <- read.csv("Cetacea_25/paralog_exons.txt", header=F)
exons_aln_error <- read.table("Cetacea_25/List_exons_error_aln_old.csv", header=T)
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_paralog$V1, ]
tab_tri_ep <- tab_tri_ep[!tab_tri_ep$Exon %in% exons_aln_error$Exon, ]
	# retrieve episodes with gof < 0.2
tab_tri_ep <- tab_tri_ep[tab_tri_ep$GoF > 0.2, ]
	# table of punctual episodes
tab_ep_ponctual <- subset(tab_tri_ep, post1 > post2 & post1 > post3)
tab_ep_ponctual <- subset(tab_ep_ponctual, tab_ep_ponctual$post1 > 0.6)
	# table of two-branches episodes
tab_ep_twobr <- subset(tab_tri_ep, post2 > post1 & post2 > post3)
	# table of all-exons episodes
tab_ep_allex <- subset(tab_tri_ep, post3 > post1 & post3 > post2)

#tab_ep_ponctual_2 = merge(tab_ep_ponctual, tab, by = c("Exon_ep", "Br_ep"))
#mean(tab_ep_ponctual_2$Br_lg)

#tab_ep_mean = merge(tab_tri_ep, tab, by = c("Exon_ep", "Br_ep"))
#mean(tab_ep_mean$Br_lg)
###récupère les tableaux des épisodes ponctuels, deux branches et tous les exons

#mean(tab$Br_lg)
## test for compensation ## FIGURE 5 ##

# NS after episodes with at least 1 NS-WS #

## WS


# load table
tab_random_NS_WS_postepNS <- read.csv("Cetacea_25/tab_NS_WS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WS_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
#j'ai changé le nom des colonnes, j'ai enlevé le dn/ds
colnames(tab_random_NS_WS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

#colnames(tab_ep_ponctual)
#colnames(tab_random_NS_WS_postepNS)

# retrive lines where there is an episode
tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode != 'YES', ]
###j'ai mis un = pour dire qu'on garde quand on a un épisode 
#tab_random_NS_WS_postepNS <- tab_random_NS_WS_postepNS[tab_random_NS_WS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepNS <- merge(tab_ep_allex, tab_random_NS_WS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

#tab_random_NS_WS_postepNS = subset(tab_random_NS_WS_postepNS, tab_random_NS_WS_postepNS$Br_asc_lg > 0.004)

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepNS <- sum(tab_random_NS_WS_postepNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepNS <- as.data.frame(colSums(tab_random_NS_WS_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepNS > distrib_nb_NS_att_WS_postepNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepNS < distrib_nb_NS_att_WS_postepNS$WS))/1000 # if deficit



## SW

# load table
tab_random_NS_SW_postepNS <- read.csv("Cetacea_25/tab_NS_SW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_SW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode != 'YES', ]
#tab_random_NS_SW_postepNS <- tab_random_NS_SW_postepNS[tab_random_NS_SW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepNS <- merge(tab_ep_allex, tab_random_NS_SW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepNS <- sum(tab_random_NS_SW_postepNS$Nb_subst_NS_SW)


# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepNS <- as.data.frame(colSums(tab_random_NS_SW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepNS > distrib_nb_NS_att_SW_postepNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepNS < distrib_nb_NS_att_SW_postepNS$SW))/1000 # if deficit
	


#SS

# load table
tab_random_NS_SS_postepNS <- read.csv("Cetacea_25/tab_NS_SS_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_SW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")

#colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br_ep', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'))
colnames(tab_random_NS_SS_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_SS'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")


# retrive lines where there is an episode
tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode != 'YES', ]
#tab_random_NS_SS_postepNS <- tab_random_NS_SS_postepNS[tab_random_NS_SS_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_SS_postepNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepNS <- merge(tab_ep_allex, tab_random_NS_SS_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepNS <- sum(tab_random_NS_SS_postepNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepNS <- as.data.frame(colSums(tab_random_NS_SS_postepNS[, 17:1016]))

colnames(distrib_nb_NS_att_SS_postepNS) <- c("SS")
	


#WW

# load table
tab_random_NS_WW_postepNS <- read.csv("Cetacea_25/tab_NS_WW_after_epNS_with_random.csv", header=T, sep=',')
#colnames(tab_random_NS_WW_postepNS) <- c(c("Exon_ep", "dNdS", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg", "Ep_uniq")
colnames(tab_random_NS_WW_postepNS) <- c(c('Exon_ep', 'Lg_seq', 'GC12', 'GC3', 'Br', 'Br_lg', 'Episode', 'Nb_subst_S_WS', 'Nb_subst_NS_WW'), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")



# retrive lines where there is an episode
tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode != 'YES', ]
#tab_random_NS_WW_postepNS <- tab_random_NS_WW_postepNS[tab_random_NS_WW_postepNS$Episode == 'YES', ]

# select a type of episode
tab_random_NS_WW_postepNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepNS <- merge(tab_ep_allex, tab_random_NS_WW_postepNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepNS <- sum(tab_random_NS_WW_postepNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepNS <- as.data.frame(colSums(tab_random_NS_WW_postepNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepNS <- nb_NS_SS_obs_tot_postepNS + nb_NS_WW_obs_tot_postepNS
# expected
tot_NS_att_SSWW_postepNS <- as.data.frame(distrib_nb_NS_att_SS_postepNS + distrib_nb_NS_att_WW_postepNS)
colnames(tot_NS_att_SSWW_postepNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepNS > tot_NS_att_SSWW_postepNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepNS < tot_NS_att_SSWW_postepNS))/1000 # if deficit




## total NS = WS+SW+SSWW

# observed
tot_NS_obs_postepNS <- nb_NS_WS_obs_tot_postepNS + nb_NS_SW_obs_tot_postepNS + nb_NS_SSWW_obs_tot_postepNS
# expected
tot_NS_att_postepNS <- as.data.frame(distrib_nb_NS_att_WS_postepNS + distrib_nb_NS_att_SW_postepNS + distrib_nb_NS_att_WW_postepNS + distrib_nb_NS_att_SS_postepNS)
colnames(tot_NS_att_postepNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepNS > tot_NS_att_postepNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepNS < tot_NS_att_postepNS$total))/1000 # if deficit


	## plots


# WS
plot_NS_WS_ponctual <- ggplot(distrib_nb_NS_att_WS_postepNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual

# SW
plot_NS_SW_ponctual <- ggplot(distrib_nb_NS_att_SW_postepNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual
# SSWW
plot_NS_SSWW_ponctual <- ggplot(tot_NS_att_SSWW_postepNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual <- ggplot(tot_NS_att_postepNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual
# merge all this plot
plot_NS_types_postepNS <- ggarrange(plot_NS_WS_ponctual, plot_NS_SW_ponctual, plot_NS_SSWW_ponctual, ncol = 3)
plot_NS_types_postepNS








# NS after episodes with no NS-WS #

## WS

# load table
tab_random_NS_WS_postepnoNS <- read.csv("Cetacea_25/tab_NS_WS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WS_postepnoNS <- tab_random_NS_WS_postepnoNS[tab_random_NS_WS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WS_obs_tot_postepnoNS <- sum(tab_random_NS_WS_postepnoNS$Nb_subst_NS_WS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WS_postepnoNS <- as.data.frame(colSums(tab_random_NS_WS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WS_postepnoNS) <- c("WS")

# p-value
(1000-sum(nb_NS_WS_obs_tot_postepnoNS > distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if excess
(1000-sum(nb_NS_WS_obs_tot_postepnoNS < distrib_nb_NS_att_WS_postepnoNS$WS))/1000 # if deficit





## SW




# load table
tab_random_NS_SW_postepnoNS <- read.csv("Cetacea_25/tab_NS_SW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SW_postepnoNS <- tab_random_NS_SW_postepnoNS[tab_random_NS_SW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SW_obs_tot_postepnoNS <- sum(tab_random_NS_SW_postepnoNS$Nb_subst_NS_SW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SW_postepnoNS <- as.data.frame(colSums(tab_random_NS_SW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SW_postepnoNS) <- c("SW")

# p-value
(1000-sum(nb_NS_SW_obs_tot_postepnoNS > distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if excess
(1000-sum(nb_NS_SW_obs_tot_postepnoNS < distrib_nb_NS_att_SW_postepnoNS$SW))/1000 # if deficit



## SS



# load table
tab_random_NS_SS_postepnoNS <- read.csv("Cetacea_25/tab_NS_SS_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_SS_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_SS"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_SS_postepnoNS <- tab_random_NS_SS_postepnoNS[tab_random_NS_SS_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_SS_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_SS_postepnoNS <- merge(tab_ep_allex, tab_random_NS_SS_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_SS_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_SS_obs_tot_postepnoNS <- sum(tab_random_NS_SS_postepnoNS$Nb_subst_NS_SS)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_SS_postepnoNS <- as.data.frame(colSums(tab_random_NS_SS_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_SS_postepnoNS) <- c("SS")



## WW




# load table
tab_random_NS_WW_postepnoNS <- read.csv("Cetacea_25/tab_NS_WW_after_epnoNS_with_random.csv", header=T, sep=',')
colnames(tab_random_NS_WW_postepnoNS) <- c(c("Exon_ep", "Lg_seq", "GC12", "GC3", "Br"," Br_lg", "Episode", "Nb_subst_S_WS", "Nb_subst_NS_WW"), as.character(seq(1,1000)), "Br_ep", "Br_asc_lg")

# retrive lines where there is an episode
tab_random_NS_WW_postepnoNS <- tab_random_NS_WW_postepnoNS[tab_random_NS_WW_postepnoNS$Episode != 'YES', ]

# select a type of episode
tab_random_NS_WW_postepnoNS <- merge(tab_ep_ponctual, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # punctual
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_twobr, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # two-branches
#tab_random_NS_WW_postepnoNS <- merge(tab_ep_allex, tab_random_NS_WW_postepnoNS, by = c("Exon_ep", "Br_ep")) # all-exons

# count nunber of episodes
tab_count_ep <- tab_random_NS_WW_postepnoNS[, c("Exon_ep", "Br_ep")]
nrow(distinct(tab_count_ep))

# sum all the NS substitutions observed in post-episode branches
nb_NS_WW_obs_tot_postepnoNS <- sum(tab_random_NS_WW_postepnoNS$Nb_subst_NS_WW)

# sum all of the NS substitutions expected in post-episode branches, for each of the 1000 randomisations (it gives a vector of 1000 values)
distrib_nb_NS_att_WW_postepnoNS <- as.data.frame(colSums(tab_random_NS_WW_postepnoNS[, 17:1016]))
colnames(distrib_nb_NS_att_WW_postepnoNS) <- c("WW")

	## SS+WW
# observed
nb_NS_SSWW_obs_tot_postepnoNS <- nb_NS_SS_obs_tot_postepnoNS + nb_NS_WW_obs_tot_postepnoNS
# expected
tot_NS_att_SSWW_postepnoNS <- as.data.frame(distrib_nb_NS_att_SS_postepnoNS + distrib_nb_NS_att_WW_postepnoNS)
colnames(tot_NS_att_SSWW_postepnoNS) <- c("SSWW")
# p-value
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS > tot_NS_att_SSWW_postepnoNS))/1000 # if excess
(1000-sum(nb_NS_SSWW_obs_tot_postepnoNS < tot_NS_att_SSWW_postepnoNS))/1000 # if deficit


## total NS = WS+SW+SSWW


# observed
tot_NS_obs_postepnoNS <- nb_NS_WS_obs_tot_postepnoNS + nb_NS_SW_obs_tot_postepnoNS + nb_NS_SSWW_obs_tot_postepnoNS
# expected
tot_NS_att_postepnoNS <- as.data.frame(distrib_nb_NS_att_WS_postepnoNS + distrib_nb_NS_att_SW_postepnoNS + distrib_nb_NS_att_WW_postepnoNS + distrib_nb_NS_att_SS_postepnoNS)
colnames(tot_NS_att_postepnoNS) <- c("total")
# p-value
(1000-sum(tot_NS_obs_postepnoNS > tot_NS_att_postepnoNS$total))/1000 # if excess
(1000-sum(tot_NS_obs_postepnoNS < tot_NS_att_postepnoNS$total))/1000 # if deficit



## plots


# WS
plot_NS_WS_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_WS_postepnoNS, aes(WS)) +
  geom_density(color="black", fill="coral2", size=0.3) +
  geom_vline(xintercept = nb_NS_WS_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10))
plot_NS_WS_ponctual_postepnoNS
# SW
plot_NS_SW_ponctual_postepnoNS <- ggplot(distrib_nb_NS_att_SW_postepnoNS, aes(SW)) +
  geom_density(color="black", fill="cornflowerblue", size=0.3) +
  geom_vline(xintercept = nb_NS_SW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SW_ponctual_postepnoNS
# SSWW
plot_NS_SSWW_ponctual_postepnoNS <- ggplot(tot_NS_att_SSWW_postepnoNS, aes(SSWW)) +
  geom_density(color="black", fill="darkseagreen", size=0.3) +
  geom_vline(xintercept = nb_NS_SSWW_obs_tot_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_SSWW_ponctual_postepnoNS
# Total (WS+SW+SSWW)
plot_NS_tot_ponctual_postepnoNS <- ggplot(tot_NS_att_postepnoNS, aes(total)) +
  geom_density(color="black", fill="plum3", size=0.3) +
  geom_vline(xintercept = tot_NS_obs_postepnoNS, color = "black", size=0.6) +
  xlab("") +
  ylab("density") +
  theme_linedraw() +
  theme(axis.title.x = element_text(size = 10), axis.title.y = element_text(size=10)) 
plot_NS_tot_ponctual_postepnoNS

# merge all this plot
plot_NS_types_postepnoNS <- ggarrange(plot_NS_WS_ponctual_postepnoNS, plot_NS_SW_ponctual_postepnoNS, plot_NS_SSWW_ponctual_postepnoNS, ncol = 3)
plot_NS_types_postepnoNS

	## final plot (Figure 5)
plot_NS_types_postep_ponctual <- ggarrange(plot_NS_types_postepNS, plot_NS_types_postepnoNS, nrow=2)
plot_NS_types_postep_ponctual

plot_NS_all_postep_ponctual <- ggarrange(plot_NS_tot_ponctual, plot_NS_tot_ponctual_postepnoNS, nrow=2)
plot_NS_all_postep_ponctual

plot_NS_postep_ponctual <- ggarrange(plot_NS_all_postep_ponctual, plot_NS_types_postep_ponctual, ncol=2, labels=c("A","B"), widths = c(2,4.5))
plot_NS_postep_ponctual








nb_NS_WS_obs_tot_postepNS
mean(distrib_nb_NS_att_WS_postepNS$WS)
sd(distrib_nb_NS_att_WS_postepNS$WS)
Z_NS_WS = (nb_NS_WS_obs_tot_postepNS - mean(distrib_nb_NS_att_WS_postepNS$WS)) / sd(distrib_nb_NS_att_WS_postepNS$WS) 

nb_NS_SW_obs_tot_postepNS
mean(distrib_nb_NS_att_SW_postepNS$SW)
sd(distrib_nb_NS_att_SW_postepNS$SW)
Z_NS_SW = (nb_NS_SW_obs_tot_postepNS-mean(distrib_nb_NS_att_SW_postepNS$SW))/sd(distrib_nb_NS_att_SW_postepNS$SW)


nb_NS_SSWW_obs_tot_postepNS
mean(tot_NS_att_SSWW_postepNS$SSWW)
sd(tot_NS_att_SSWW_postepNS$SSWW)
Z_NS_SSWW = (nb_NS_SSWW_obs_tot_postepNS-mean(tot_NS_att_SSWW_postepNS$SSWW))/sd(tot_NS_att_SSWW_postepNS$SSWW)


tot_NS_obs_postepNS
mean(tot_NS_att_postepNS$total)
sd(tot_NS_att_postepNS$total)
Z_NS_tot = (tot_NS_obs_postepNS-mean(tot_NS_att_postepNS$total))/sd(tot_NS_att_postepNS$total)




nb_NS_WS_obs_tot_postepnoNS
mean(distrib_nb_NS_att_WS_postepnoNS$WS)
sd(distrib_nb_NS_att_WS_postepnoNS$WS)

Z_no_NS_WS = (nb_NS_WS_obs_tot_postepnoNS-mean(distrib_nb_NS_att_WS_postepnoNS$WS))/sd(distrib_nb_NS_att_WS_postepnoNS$WS)

nb_NS_SW_obs_tot_postepnoNS
mean(distrib_nb_NS_att_SW_postepnoNS$SW)
sd(distrib_nb_NS_att_SW_postepnoNS$SW)
Z_no_NS_SW = (nb_NS_SW_obs_tot_postepnoNS-mean(distrib_nb_NS_att_SW_postepnoNS$SW))/sd(distrib_nb_NS_att_SW_postepnoNS$SW)

nb_NS_SSWW_obs_tot_postepnoNS
mean(tot_NS_att_SSWW_postepnoNS$SSWW)
sd(tot_NS_att_SSWW_postepnoNS$SSWW)
Z_no_NS_SSWW = (nb_NS_SSWW_obs_tot_postepnoNS-mean(tot_NS_att_SSWW_postepnoNS$SSWW))/sd(tot_NS_att_SSWW_postepnoNS$SSWW)

tot_NS_obs_postepnoNS
mean(tot_NS_att_postepnoNS$total)
sd(tot_NS_att_postepnoNS$total)

Z_no_NS_tot = (tot_NS_obs_postepnoNS-mean(tot_NS_att_postepnoNS$total))/sd(tot_NS_att_postepnoNS$total)

df_Z_scores_Cetacea_25_all <- data.frame(
  row.names = c("NS", "no_NS"),
  WS = c(Z_NS_WS, Z_no_NS_WS),
  SW = c(Z_NS_SW, Z_no_NS_SW),
  SSWW = c(Z_NS_SSWW, Z_no_NS_SSWW),
  total = c(Z_NS_tot, Z_no_NS_tot)
)





# Melt avec les noms de ligne comme variable "Condition"
df_points_Cetacea_25_all <- melt(df_Z_scores_Cetacea_25_all, varnames = c("Condition", "Transition"))
df_points_Cetacea_25_all$Condition <- rep(rownames(df_Z_scores_Cetacea_25_all), times = ncol(df_Z_scores_Cetacea_25_all))

plot_Cetacea_25_all = ggplot(df_points_Cetacea_25_all, aes(x = variable, y = value, color = Condition)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  ylab("Z-score") +
  xlab("Type de transition") +
  ggtitle("Z-scores NS vs no_NS") +
  scale_color_manual(values = c("steelblue", "tomato")) +
  theme_minimal()

plot_Cetacea_25_all


```

##Z score selon le nb d'espèces 
```{r}
###plot des subst tot
subst_tot_df <- data.frame(
  Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp"),
  total_subst = c(Nb_totaL_subst_15, Nb_totaL_subst_25, Nb_totaL_subst_37, Nb_totaL_subst_all, Nb_totaL_subst_Murini, Nb_totaL_subst_Rattini, Nb_totaL_subst_Cetacea, Nb_totaL_subst_Cetacea_25)
)

df_points_15_all$nb_esp <- 15
df_points_25_all$nb_esp <- 25
df_points_37_all$nb_esp <- 37
df_points_48_all$nb_esp <- 48
# Fusionner tous les data frames

# Fusionner tous les data frames en les empilant verticalement
df_all_all_lg <- rbind(df_points_15_all, df_points_25_all, df_points_37_all, df_points_48_all)

##15 sp
df_15_WS_all = subset(df_points_15_all, df_points_15_all$variable == "WS")
df_15_WS_all <- df_15_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_SW_all = subset(df_points_15_all, df_points_15_all$variable == "SW")
df_15_SW_all <- df_15_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_SSWW_all = subset(df_points_15_all, df_points_15_all$variable == "SSWW")
df_15_SSWW_all <- df_15_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_15_total_all = subset(df_points_15_all, df_points_15_all$variable == "total")
df_15_total_all <- df_15_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

#25sp

df_25_WS_all = subset(df_points_25_all, df_points_25_all$variable == "WS")
df_25_WS_all <- df_25_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_SW_all = subset(df_points_25_all, df_points_25_all$variable == "SW")
df_25_SW_all <- df_25_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_SSWW_all = subset(df_points_25_all, df_points_25_all$variable == "SSWW")
df_25_SSWW_all <- df_25_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_25_total_all = subset(df_points_25_all, df_points_25_all$variable == "total")
df_25_total_all <- df_25_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

#37
df_37_WS_all = subset(df_points_37_all, df_points_37_all$variable == "WS")
df_37_WS_all <- df_37_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_SW_all = subset(df_points_37_all, df_points_37_all$variable == "SW")
df_37_SW_all <- df_37_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_SSWW_all = subset(df_points_37_all, df_points_37_all$variable == "SSWW")
df_37_SSWW_all <- df_37_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_37_total_all = subset(df_points_37_all, df_points_37_all$variable == "total")
df_37_total_all <- df_37_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

##48sp
df_48_WS_all = subset(df_points_48_all, df_points_48_all$variable == "WS")
df_48_WS_all <- df_48_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_SW_all = subset(df_points_48_all, df_points_48_all$variable == "SW")
df_48_SW_all <- df_48_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_SSWW_all = subset(df_points_48_all, df_points_48_all$variable == "SSWW")
df_48_SSWW_all <- df_48_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_48_total_all = subset(df_points_48_all, df_points_48_all$variable == "total")
df_48_total_all <- df_48_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


###Murini
df_Murini_WS_all = subset(df_points_Murini_all, df_points_Murini_all$variable == "WS")
df_Murini_WS_all <- df_Murini_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_SW_all = subset(df_points_Murini_all, df_points_Murini_all$variable == "SW")
df_Murini_SW_all <- df_Murini_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_SSWW_all = subset(df_points_Murini_all, df_points_Murini_all$variable == "SSWW")
df_Murini_SSWW_all <- df_Murini_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Murini_total_all = subset(df_points_Murini_all, df_points_Murini_all$variable == "total")
df_Murini_total_all <- df_Murini_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


###Rattini

df_Rattini_WS_all = subset(df_points_Rattini_all, df_points_Rattini_all$variable == "WS")
df_Rattini_WS_all <- df_Rattini_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_SW_all = subset(df_points_Rattini_all, df_points_Rattini_all$variable == "SW")
df_Rattini_SW_all <- df_Rattini_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_SSWW_all = subset(df_points_Rattini_all, df_points_Rattini_all$variable == "SSWW")
df_Rattini_SSWW_all <- df_Rattini_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Rattini_total_all = subset(df_points_Rattini_all, df_points_Rattini_all$variable == "total")
df_Rattini_total_all <- df_Rattini_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

####Cétacea

df_Cetacea_WS_all = subset(df_points_Cetacea_all, df_points_Cetacea_all$variable == "WS")
df_Cetacea_WS_all <- df_Cetacea_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_SW_all = subset(df_points_Cetacea_all, df_points_Cetacea_all$variable == "SW")
df_Cetacea_SW_all <- df_Cetacea_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_SSWW_all = subset(df_points_Cetacea_all, df_points_Cetacea_all$variable == "SSWW")
df_Cetacea_SSWW_all <- df_Cetacea_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_total_all = subset(df_points_Cetacea_all, df_points_Cetacea_all$variable == "total")
df_Cetacea_total_all <- df_Cetacea_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  


####Cétacea 25 sp

df_Cetacea_25_WS_all = subset(df_points_Cetacea_25_all, df_points_Cetacea_25_all$variable == "WS")
df_Cetacea_25_WS_all <- df_Cetacea_25_WS_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_SW_all = subset(df_points_Cetacea_25_all, df_points_Cetacea_25_all$variable == "SW")
df_Cetacea_25_SW_all <- df_Cetacea_25_SW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_SSWW_all = subset(df_points_Cetacea_25_all, df_points_Cetacea_25_all$variable == "SSWW")
df_Cetacea_25_SSWW_all <- df_Cetacea_25_SSWW_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)   

df_Cetacea_25_total_all = subset(df_points_Cetacea_25_all, df_points_Cetacea_25_all$variable == "total")
df_Cetacea_25_total_all <- df_Cetacea_25_total_all %>%
  spread(Condition, value) %>%  # Transformer les conditions NS et no_NS en colonnes séparées
  mutate(diff = NS - no_NS)  

####merge les tableaux 



diff_WS_all <- rbind(
  df_15_WS_all %>% mutate(nb_esp = "15"),
  df_25_WS_all %>% mutate(nb_esp = "25"),
  df_37_WS_all %>% mutate(nb_esp = "37"),
  df_48_WS_all %>% mutate(nb_esp = "48"),
  df_Murini_WS_all %>% mutate(nb_esp = "10"),
  df_Rattini_WS_all %>% mutate(nb_esp = "19"),
  df_Cetacea_WS_all %>% mutate(nb_esp = "75"),
  df_Cetacea_25_WS_all %>% mutate(nb_esp = "25")
)


diff_SW_all <- rbind(
  df_15_SW_all %>% mutate(nb_esp = "15"),
  df_25_SW_all %>% mutate(nb_esp = "25"),
  df_37_SW_all %>% mutate(nb_esp = "37"),
  df_48_SW_all %>% mutate(nb_esp = "48"),
  df_Murini_SW_all %>% mutate(nb_esp = "10"),
  df_Rattini_SW_all %>% mutate(nb_esp = "19"),
  df_Cetacea_SW_all %>% mutate(nb_esp = "75"),
  df_Cetacea_25_SW_all %>% mutate(nb_esp = "25")
)


diff_SSWW_all <- rbind(
  df_15_SSWW_all %>% mutate(nb_esp = "15"),
  df_25_SSWW_all %>% mutate(nb_esp = "25"),
  df_37_SSWW_all %>% mutate(nb_esp = "37"),
  df_48_SSWW_all %>% mutate(nb_esp = "48"),
  df_Murini_SSWW_all %>% mutate(nb_esp = "10"),
  df_Rattini_SSWW_all %>% mutate(nb_esp = "19"),
  df_Cetacea_SSWW_all %>% mutate(nb_esp = "75"),
  df_Cetacea_25_SSWW_all %>% mutate(nb_esp = "25")
)




diff_total_all <- rbind(
  df_15_total_all %>% mutate(nb_esp = "15"),
  df_25_total_all %>% mutate(nb_esp = "25"),
  df_37_total_all %>% mutate(nb_esp = "37"),
  df_48_total_all %>% mutate(nb_esp = "48"),
  df_Murini_total_all %>% mutate(nb_esp = "10"),
  df_Rattini_total_all %>% mutate(nb_esp = "19"),
  df_Cetacea_total_all %>% mutate(nb_esp = "75"),
  df_Cetacea_25_total_all %>% mutate(nb_esp = "25")
)



###changer le nom des lignes et de la colonne
diff_WS_all_all_lg <- diff_WS_all[, c(1:5)]  
diff_WS_all_all_lg$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")





diff_SW_all_all_lg <- diff_SW_all[, c(1:5)]  
diff_SW_all_all_lg$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")

diff_SSWW_all_all_lg <- diff_SSWW_all[, c(1:5)]  
diff_SSWW_all_all_lg$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")

diff_total_all_all_lg <- diff_total_all[, c(1:5)]  
diff_total_all_all_lg$Groupe = c("15_sp_Hydro", "25_sp_Hydro", "37_sp_Hydro", "48_sp_Hydro", "Murini", "Rattini", "Cetacea", "Cetacea_25_sp")


data_subst_WS_all_lg = merge(diff_WS_all_all_lg, subst_tot_df, by = c("Groupe"))
data_subst_WS_all_lg$Groupe = c("Hydromyini", "Hydromyini", "Hydromyini", "Hydromyini", "Cetacea", "Cetacea", "Murini", "Rattini")
data_subst_SW_all_lg = merge(diff_SW_all_all_lg, subst_tot_df, by = c("Groupe"))
data_subst_SW_all_lg$Groupe = c("Hydromyini", "Hydromyini", "Hydromyini", "Hydromyini", "Cetacea", "Cetacea", "Murini", "Rattini")
data_subst_SSWW_all_lg = merge(diff_SSWW_all_all_lg, subst_tot_df, by = c("Groupe"))
data_subst_SSWW_all_lg$Groupe = c("Hydromyini", "Hydromyini", "Hydromyini", "Hydromyini", "Cetacea", "Cetacea", "Murini", "Rattini")
data_subst_total_all_lg = merge(diff_total_all_all_lg, subst_tot_df, by = c("Groupe"))
data_subst_total_all_lg$Groupe = c("Hydromyini", "Hydromyini", "Hydromyini", "Hydromyini", "Cetacea", "Cetacea", "Murini", "Rattini")



machin = read.csv("nb_esp_fasta.csv", sep ="\t", header =F)


# Définir une palette de couleurs 
couleurs_Groupes <- c(
  "Hydromyini" = "#1b9e77",
  "Murini"       = "#d95f02",
  "Rattini"      = "#BA8383",
  "Cetacea"      = "#107FB1"
)

# Définir une palette de formes (shapes) : valeurs entre 0 et 25
formes_Groupes <- c(
  "Hydromyini" = 15,  
  "Murini"     = 17,  
  "Rattini"    = 19,  
  "Cetacea"    = 18   
)


plot_all_lg_WS=ggplot(data_subst_WS_all_lg, aes(x = (total_subst), y = diff, color = Groupe, shape = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(aes(group=1),method = "lm", se = T, color = "black", size = 0.5) +
  scale_color_manual(values = couleurs_Groupes) +
  scale_shape_manual(values = formes_Groupes) +
  ylab("sélection compensatoire pour les substitutions WS") +
  xlab("Nombre de substitutions") +
#  ggtitle("Z-score : NS - no_NS pour WS selon le nombre d'espèces") +
  geom_text(aes(label = nb_esp), vjust = 1.5, hjust = 1.5)+
  theme_minimal()

plot_all_lg_WS

ggsave("diff_Z_score_all_sp/plot_all_lg_WS.png", plot = plot_all_lg_WS, width = 8, height = 6, dpi = 300)




plot_all_lg_SW=ggplot(data_subst_SW_all_lg, aes(x = (total_subst), y = diff, color = Groupe, shape = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(aes(group=1),method = "lm", se = T, color = "black", size = 0.5) +
  scale_color_manual(values = couleurs_Groupes) +
  scale_shape_manual(values = formes_Groupes) +
 ylab("sélection compensatoire pour les substitutions SW") +
  xlab("Nombre de substitutions") +
#  ggtitle("Z-score : NS - no_NS pour SW selon le nombre d'espèces") +
  geom_text(aes(label = nb_esp), vjust = 0.2, hjust = 1.5)+
  theme_minimal()
plot_all_lg_SW
ggsave("diff_Z_score_all_sp/plot_all_lg_SW.png", plot = plot_all_lg_SW, width = 8, height = 6, dpi = 300)


plot_all_lg_SSWW=ggplot(data_subst_SSWW_all_lg, aes(x = (total_subst), y = diff, color = Groupe, shape = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(aes(group=1),method = "lm", se = T, color = "black", size = 0.5) +
  scale_shape_manual(values = formes_Groupes) +
  scale_color_manual(values = couleurs_Groupes) +
  ylab("sélection compensatoire pour les substitutions SSWW") +
  xlab("Nombre de substitutions") +
# ggtitle("Z-score : NS - no_NS pour SSWW selon le nombre d'espèces") +
  geom_text(aes(label = nb_esp), vjust = -1, hjust = 1)+
  theme_minimal()

plot_all_lg_SSWW

ggsave("diff_Z_score_all_sp/plot_all_lg_SSWW.png", plot = plot_all_lg_SSWW, width = 8, height = 6, dpi = 300)

plot_all_lg_total=ggplot(data_subst_total_all_lg, aes(x = (total_subst), y = diff, color = Groupe, shape = Groupe)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(aes(group=1),method = "lm", se = T, color = "black", size = 0.5) +
  scale_shape_manual(values = formes_Groupes) +
  geom_text(aes(label = nb_esp), vjust = 0.2, hjust = 1.5)+
  ylab("sélection compensatoire pour toutes les substitutions") +
  xlab("Nombre de substitutions") +
#  ggtitle("sélection compensatoire") +
  scale_color_manual(values = couleurs_Groupes) +
  theme_minimal()

plot_all_lg_total
ggsave("diff_Z_score_all_sp/plot_all_lg_total.png", plot = plot_all_lg_total, width = 8, height = 6, dpi = 300)





















data_subst_WS_all_lg_all_subst = data_subst_WS_all_lg
data_subst_SW_all_lg_all_subst = data_subst_SW_all_lg
data_subst_SSWW_all_lg_all_subst = data_subst_SSWW_all_lg
data_subst_total_all_lg_all_subst = data_subst_total_all_lg



# Combiner toutes les données ensemble
data_all_subst <- bind_rows(
  data_subst_WS_all_lg_all_subst,
  data_subst_SW_all_lg_all_subst,
  data_subst_SSWW_all_lg_all_subst,
  data_subst_total_all_lg_all_subst
)

# Définir une palette de couleurs pour les types de substitutions
couleurs_types <- c(
  "WS" = "coral2",
  "SW" = "cornflowerblue",
  "SSWW" = "darkseagreen",
  "total" = "plum3"
)


# Plot
plot_all_categories <- ggplot(data_all_subst, aes(x = total_subst, y = diff, color = variable)) +
  geom_point(size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray40") +
  geom_smooth(method = "lm", se = F, size = 0.8) +
  scale_color_manual(values = couleurs_types) +
  ylab("sélection compensatoire") +
  xlab("Nombre de substitutions") +
#  ggtitle("Z-score NS - no_NS pour WS, SW, SSWW et Total") +
  theme_minimal() +
  theme(legend.position = "top")

plot_all_categories

ggsave("diff_Z_score_all_sp/plot_all_categories.png", plot = plot_all_categories, width = 8, height = 6, dpi = 300)



```





####les plots avec et sans longueurs de branches 


```{r}
plot_Z_score_lg_WS <- ggarrange(plot_all_lg_WS, plot_subst_WS_lg_inf_0.006, nrow=2)
plot_Z_score_lg_WS


plot_Z_score_lg_SW <- ggarrange(plot_all_lg_SW, plot_subst_SW_lg_inf_0.006, nrow=2)
plot_Z_score_lg_SW

plot_Z_score_lg_SSWW <- ggarrange(plot_all_lg_SSWW, plot_subst_SSWW_lg_inf_0.006, nrow=2)
plot_Z_score_lg_SSWW

plot_Z_score_lg_total <- ggarrange(plot_all_lg_total, plot_subst_total_lg_inf_0.006, nrow=2)
plot_Z_score_lg_total

```




###mettre les  Z score des branches imposées et non imposées 
```{r}
# Sélectionner les colonnes 1, 2 et 6
data_subst_WS_comp = data_subst_WS[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_WS_comp) = c("Groupe", "variable", "diff_lg_inf" ,"total_subst")

# Sélectionner les colonnes 1, 2 et 6
data_subst_SW_comp = data_subst_SW[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_SW_comp) = c("Groupe", "variable","diff_lg_inf" , "total_subst")


# Sélectionner les colonnes 1, 2 et 6
data_subst_SSWW_comp = data_subst_SSWW[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_SSWW_comp) = c("Groupe", "variable","diff_lg_inf" , "total_subst")

# Sélectionner les colonnes 1, 2 et 6
data_subst_total_comp = data_subst_total[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_total_comp) = c("Groupe", "variable","diff_lg_inf" , "total_subst")




# Sélectionner les colonnes 1, 2 et 6
data_subst_WS_all_lg_comp = data_subst_WS_all_lg[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_WS_all_lg_comp ) = c("Groupe", "variable", "diff_all_lg" ,"total_subst")

# Sélectionner les colonnes 1, 2 et 6
data_subst_SW_all_lg_comp = data_subst_SW_all_lg[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_SW_all_lg_comp ) = c("Groupe", "variable","diff_all_lg" , "total_subst")


# Sélectionner les colonnes 1, 2 et 6
data_subst_SSWW_all_lg_comp = data_subst_SSWW_all_lg[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_SSWW_all_lg_comp) = c("Groupe", "variable", "diff_all_lg", "total_subst")

# Sélectionner les colonnes 1, 2 et 6
data_subst_total_all_lg_comp = data_subst_total_all_lg[, c(1, 2,5, 6)]
# Renommer correctement les colonnes
colnames(data_subst_total_all_lg_comp ) = c("Groupe", "variable", "diff_all_lg","total_subst")



data_diff_lg_WS = merge(data_subst_WS_all_lg_comp,data_subst_WS_comp, by = c("Groupe", "variable", "total_subst") )

data_diff_lg_SW = merge(data_subst_SW_all_lg_comp,data_subst_SW_comp, by = c("Groupe", "variable", "total_subst") )

data_diff_lg_SSWW = merge(data_subst_SSWW_all_lg_comp,data_subst_SSWW_comp, by = c("Groupe", "variable", "total_subst") )

data_diff_lg_total = merge(data_subst_total_all_lg_comp,data_subst_total_comp, by = c("Groupe", "variable", "total_subst") )






# D'abord, restructurer les données au format long pour les variables diff_all_lg et diff_lg_inf
data_long <- tidyr::pivot_longer(
  data_diff_lg_total,
  cols = c(diff_all_lg, diff_lg_inf),
  names_to = "type_diff",
  values_to = "valeur_diff"
)


data_long <- data_long %>% arrange(Groupe, total_subst)
# Créer le graphique
ggplot(data_long, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(size = 3) +  # Points pour chaque Groupe
  geom_line(aes(group = interaction(Groupe, type_diff), linetype = type_diff), size = 0.8) +
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                              "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                              "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                              "Murini" = "cyan", "Rattini" = "yellow")) +
  scale_linetype_manual(values = c("diff_all_lg" = "solid", "diff_lg_inf" = "dashed")) +
  labs(
    y = "Différence",
    x = "Nombre de substitutions",
    title = "Comparaison des différences selon le nombre de substitutions",
    linetype = "Type de différence"
  ) +
  theme_minimal() +
  theme(legend.position = "top")











# Transforme ton tableau en long format
data_subst_long_WS <- data_diff_lg_WS %>%
  pivot_longer(
    cols = c(diff_all_lg, diff_lg_inf),
    names_to = "type_diff",
    values_to = "valeur_diff"
  )

ggplot(data_subst_long_WS, aes(x = total_subst, y = valeur_diff, color = type_diff, group = type_diff)) +
  geom_point(size = 3) +
  geom_line(size = 1) +
  scale_color_manual(values = c("diff_all_lg" = "blue", "diff_lg_inf" = "red")) +
  xlab("Nombre de substitutions") +
  ylab("Différence Z-score") +
  ggtitle("WS : longueurs toutes vs longues inférieures") +
  theme_minimal()

# Transforme ton tableau en long format
data_subst_long_SW <- data_diff_lg_SW %>%
  pivot_longer(
    cols = c(diff_all_lg, diff_lg_inf),
    names_to = "type_diff",
    values_to = "valeur_diff"
  )

ggplot(data_subst_long_SW, aes(x = total_subst, y = valeur_diff, color = type_diff, group = type_diff)) +
  geom_point(size = 3) +
  geom_line(size = 1) +
  scale_color_manual(values = c("diff_all_lg" = "blue", "diff_lg_inf" = "red")) +
  xlab("Nombre de substitutions") +
  ylab("Différence Z-score") +
  ggtitle("SW : longueurs toutes vs longues inférieures") +
  theme_minimal()



data_subst_long_SSWW <- data_diff_lg_SSWW %>%
  pivot_longer(
    cols = c(diff_all_lg, diff_lg_inf),
    names_to = "type_diff",
    values_to = "valeur_diff"
  )

ggplot(data_subst_long_SSWW, aes(x = total_subst, y = valeur_diff, color = type_diff, group = type_diff)) +
  geom_point(size = 3) +
  geom_line(size = 1) +
  scale_color_manual(values = c("diff_all_lg" = "blue", "diff_lg_inf" = "red")) +
  xlab("Nombre de substitutions") +
  ylab("Différence Z-score") +
  ggtitle("SSWW : longueurs toutes vs longues inférieures") +
  theme_minimal()



data_subst_long_total <- data_diff_lg_total %>%
  pivot_longer(
    cols = c(diff_all_lg, diff_lg_inf),
    names_to = "type_diff",
    values_to = "valeur_diff"
  )

ggplot(data_subst_long_total, aes(x = total_subst, y = valeur_diff, color = type_diff, group = type_diff)) +
  geom_point(size = 3) +
  geom_line(size = 1) +
  scale_color_manual(values = c("diff_all_lg" = "blue", "diff_lg_inf" = "red")) +
  xlab("Nombre de substitutions") +
  ylab("Différence Z-score") +
  ggtitle("total : longueurs toutes vs longues inférieures") +
  theme_minimal()



ggplot(data_subst_long_WS, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +
  geom_segment(
    data = pivot_wider(data_subst_long_WS, names_from = type_diff, values_from = valeur_diff),
    aes(
      x = total_subst,
      xend = total_subst,
      y = diff_all_lg,
      yend = diff_lg_inf,
      color = Groupe
    ),
    inherit.aes = FALSE,
    linetype = "dashed",
    size = 0.8
  ) +
  ylab("Différence Z-score") +
  xlab("Nombre de substitutions") +
  theme_minimal() +
  ggtitle("WS : Comparaison des différences selon le nombre de substitutions")

ggplot(data_subst_long_SW, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +
  geom_segment(
    data = pivot_wider(data_subst_long_WS, names_from = type_diff, values_from = valeur_diff),
    aes(
      x = total_subst,
      xend = total_subst,
      y = diff_all_lg,
      yend = diff_lg_inf,
      color = Groupe
    ),
    inherit.aes = FALSE,
    linetype = "dashed",
    size = 0.8
  ) +
  ylab("Différence Z-score") +
  xlab("Nombre de substitutions") +
  theme_minimal() +
  ggtitle("SW : Comparaison des différences selon le nombre de substitutions")


ggplot(data_subst_long_SSWW, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +
  geom_segment(
    data = pivot_wider(data_subst_long_WS, names_from = type_diff, values_from = valeur_diff),
    aes(
      x = total_subst,
      xend = total_subst,
      y = diff_all_lg,
      yend = diff_lg_inf,
      color = Groupe
    ),
    inherit.aes = FALSE,
    linetype = "dashed",
    size = 0.8
  ) +
  ylab("Différence Z-score") +
  xlab("Nombre de substitutions") +
  theme_minimal() +
  ggtitle("SSWW : Comparaison des différences selon le nombre de substitutions")


ggplot(data_subst_long_total, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +
  geom_segment(
    data = pivot_wider(data_subst_long_WS, names_from = type_diff, values_from = valeur_diff),
    aes(
      x = total_subst,
      xend = total_subst,
      y = diff_all_lg,
      yend = diff_lg_inf,
      color = Groupe
    ),
    inherit.aes = FALSE,
    linetype = "dashed",
    size = 0.8
  ) +
  ylab("Différence Z-score") +
  xlab("Nombre de substitutions") +
  theme_minimal() +
  ggtitle("total : Comparaison des différences selon le nombre de substitutions")





ggplot(data_subst_long_WS, aes(x = total_subst, y = valeur_diff, color = Groupe, group = type_diff)) +
  geom_point(size = 3) +  # Points pour chaque Groupe
  geom_line(aes(group = type_diff), size = 1) +  # Relier les points pour chaque type_diff
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                                "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                                "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                                "Murini" = "cyan", "Rattini" = "yellow")) +  # Choix des couleurs par Groupe
  ylab("Différence Z-score") +
  xlab("Nombre de substitutions") +
  theme_minimal() +
  ggtitle("Comparaison des différences selon le nombre de substitutions") +
  theme(legend.position = "top")



ggplot(data_subst_long_WS, aes(x = total_subst, y = valeur_diff, color = Groupe, group = type_diff)) +
  geom_point(size = 3) +  # Points pour chaque Groupe
  geom_line(aes(group = type_diff), size = 0.5, color = "black") +  # Relier les points pour chaque type_diff en noir
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                                "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                                "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                                "Murini" = "cyan", "Rattini" = "yellow")) +  # Choix des couleurs par Groupe
  ylab("Différence Z-score") +
  xlab("Nombre de substitutions") +
  theme_minimal() +
  ggtitle("Comparaison des différences selon le nombre de substitutions") +
  theme(legend.position = "top")














# Créer le graphique avec formes différentes
ggplot(data_subst_long_WS, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +  # Utiliser shape pour les différents types
  geom_smooth(method = "lm", se = F, color = "lightgreen") +  # Ajout de la régression linéaire

  geom_line(aes(group = paste(Groupe, type_diff)), size = 0.6) +  # Garder les lignes pour relier les points
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                             "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                             "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                             "Murini" = "cyan", "Rattini" = "yellow")) +
  scale_shape_manual(values = c("diff_all_lg" = 17, "diff_lg_inf" = 15)) +  # 17 = triangle, 15 = carré
  labs(
    y = "Différence",
    x = "Nombre de substitutions",
    title = "WS : Comparaison des différences selon le nombre de substitutions",
    shape = "Type de différence"
  ) +
  theme_minimal() +
  theme(legend.position = "top")







# Créer le graphique avec formes différentes et régressions linéaires par type
ggplot(data_subst_long_WS, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +  # Utiliser shape pour les différents types
  geom_smooth(aes(linetype = type_diff, group = type_diff), method = "lm", se = FALSE, color = "black", size = 0.8) +  # Régressions linéaires distinctes par type_diff
  geom_line(aes(group = paste(Groupe, type_diff)), size = 0.6) +  # Garder les lignes pour relier les points
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                             "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                             "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                             "Murini" = "cyan", "Rattini" = "yellow")) +
  scale_shape_manual(values = c("diff_all_lg" = 17, "diff_lg_inf" = 15)) +  # 17 = triangle, 15 = carré
  scale_linetype_manual(values = c("diff_all_lg" = "solid", "diff_lg_inf" = "dashed")) +  # Types de lignes pour les régressions
  labs(
    y = "Différence",
    x = "Nombre de substitutions",
    title = "WS : Comparaison des différences selon le nombre de substitutions",
    shape = "Type de différence"
  ) +
  theme_minimal() +
  theme(legend.position = "top")



# Créer le graphique avec formes différentes et régressions linéaires par type
ggplot(data_subst_long_SW, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +  # Utiliser shape pour les différents types
  geom_smooth(aes(linetype = type_diff, group = type_diff), method = "lm", se = FALSE, color = "black", size = 0.8) +  # Régressions linéaires distinctes par type_diff
  geom_line(aes(group = paste(Groupe, type_diff)), size = 0.6) +  # Garder les lignes pour relier les points
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                             "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                             "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                             "Murini" = "cyan", "Rattini" = "yellow")) +
  scale_shape_manual(values = c("diff_all_lg" = 17, "diff_lg_inf" = 15)) +  # 17 = triangle, 15 = carré
  scale_linetype_manual(values = c("diff_all_lg" = "solid", "diff_lg_inf" = "dashed")) +  # Types de lignes pour les régressions
  labs(
    y = "Différence",
    x = "Nombre de substitutions",
    title = "SW : Comparaison des différences selon le nombre de substitutions",
    shape = "Type de différence"
  ) +
  theme_minimal() +
  theme(legend.position = "top")





# Créer le graphique avec formes différentes et régressions linéaires par type
ggplot(data_subst_long_SSWW, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +  # Utiliser shape pour les différents types
  geom_smooth(aes(linetype = type_diff, group = type_diff), method = "lm", se = FALSE, color = "black", size = 0.8) +  # Régressions linéaires distinctes par type_diff
  geom_line(aes(group = paste(Groupe, type_diff)), size = 0.6) +  # Garder les lignes pour relier les points
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                             "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                             "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                             "Murini" = "cyan", "Rattini" = "yellow")) +
  scale_shape_manual(values = c("diff_all_lg" = 17, "diff_lg_inf" = 15)) +  # 17 = triangle, 15 = carré
  scale_linetype_manual(values = c("diff_all_lg" = "solid", "diff_lg_inf" = "dashed")) +  # Types de lignes pour les régressions
  labs(
    y = "Différence",
    x = "Nombre de substitutions",
    title = "SSWW : Comparaison des différences selon le nombre de substitutions",
    shape = "Type de différence"
  ) +
  theme_minimal() +
  theme(legend.position = "top")


# Créer le graphique avec formes différentes et régressions linéaires par type
ggplot(data_subst_long_total, aes(x = total_subst, y = valeur_diff, color = Groupe)) +
  geom_point(aes(shape = type_diff), size = 3) +  # Utiliser shape pour les différents types
  geom_smooth(aes(linetype = type_diff, group = type_diff), method = "lm", se = FALSE, color = "black", size = 0.8) +  # Régressions linéaires distinctes par type_diff
  geom_line(aes(group = paste(Groupe, type_diff)), size = 0.6) +  # Garder les lignes pour relier les points
  scale_color_manual(values = c("15_sp_Hydro" = "blue", "25_sp_Hydro" = "red", 
                             "37_sp_Hydro" = "green", "48_sp_Hydro" = "purple", 
                             "Cetacea" = "orange", "Cetacea_25_sp" = "pink", 
                             "Murini" = "cyan", "Rattini" = "yellow")) +
  scale_shape_manual(values = c("diff_all_lg" = 17, "diff_lg_inf" = 15)) +  # 17 = triangle, 15 = carré
  scale_linetype_manual(values = c("diff_all_lg" = "solid", "diff_lg_inf" = "dashed")) +  # Types de lignes pour les régressions
  labs(
    y = "Différence",
    x = "Nombre de substitutions",
    title = "total : Comparaison des différences selon le nombre de substitutions",
    shape = "Type de différence"
  ) +
  theme_minimal() +
  theme(legend.position = "top")
```


